---
title: Sex Differences in the Impact of Prenatal Adverse Exposures on Children's Brain Structure - Implications for Cognition and Behavior Problems
author: "Euclides J. de Mendon√ßa Filho"
output:
  html_document:
    toc: yes
    theme: united
editor_options: 
  chunk_output_type: console
---

# Loading packages and functions

```{r, echo=FALSE}
setwd("/data1/meaneylab/euclides/3_ABCD cohort/Prenatal_Adversity_Univariate/Analysis_Update/")
file_path          <- "/data1/projects/ABCD_5.1/core/"
gene_path          <- "/data1/projects/genodata/ABCD/ABCD_5/genomics_sample03/imputed/PCA/ABCD_5_UNRELATED_subjects_PCA_projection_on_FULL_without_MHC_15Apr2024/"
brainvol.gwas.path <- "/data1/projects/cortexjobs/PRS_CS_score/d_prscs/jobs/Score_for_Euclides/Brain_volume_meta_analysis_BV_Jansenetal_2020_GWAS_and_ABCD5/3_Prepare_GWAS_using_PRSCS_and_calculate_score_using_PRSice/"
birthweight_path   <- '/data1/projects/ABCD/Analysis/ABC Scores/'
mri.labels         <- read.csv("abcd_smrip101_definitions.csv")


```


```{r setup, message=FALSE}
library(dplyr)      # for data manipulation
library(psych)      # for descriptives
library(ggplot2)    # for plotting
library(gridExtra)  # for arranging plots
library(kableExtra) # for creating tables
library(tidyr)      # for data manipulation
library(xlsx)       # for reading and writing xlsx files
library(tableone)   # for descriptives
library(extrafont)

select   <- dplyr::select
describe <- psych::describe
options(knitr.kable.NA = '')

```

# Loading datasets and computing study variables

```{r Preparing dataset, cache=TRUE}

#### ABCD MRI Info ####
mri_info            <- read.csv(paste0(file_path,"imaging/mri_y_adm_info.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Cortical Volume ####
mri_y_smr_vol_dsk   <- read.csv(paste0(file_path,"imaging/mri_y_smr_vol_dsk.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Cortical thickness ####
mriCortThick        <- read.csv(paste0(file_path,"imaging/mri_y_smr_thk_dsk.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Cortical Area ####
mriCortArea         <- read.csv(paste0(file_path,"imaging/mri_y_smr_area_dsk.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Sulcal Depth ####
mriSulcalDepth      <- read.csv(paste0(file_path,"imaging/mri_y_smr_sulc_dsk.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Freesurfer reconstruction quality control measures ####
fsurf_QC            <- read.csv(paste0(file_path,"imaging/mri_y_qc_man_fsurf.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Elyounssi and Roffman (2023) quality control ####
mri_y_qc_incl       <- read.csv(paste0(file_path,"imaging/mri_y_qc_incl.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

# sMRI morphometric and image intensity measures
mri_subCortVol_aseg <- read.csv(paste0(file_path,"imaging/mri_y_smr_vol_aseg.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')


#### ABCD Screener ####
p_screen           <- read.csv(paste0(file_path,"abcd-general/abcd_p_screen.csv"))

#### ABCD Parent Demographics Survey ####
pdem02            <- read.csv(paste0(file_path,"abcd-general/abcd_p_demo.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

# Sex at birth
# demo_sex_v2 - What sex was the child assigned at birth, on the original birth certificate?
#1 = Male; 2 = Female; 3 = Intersex-Male; 4 = Intersex-Female; 999 = Don't know ; 777 = Refuse to answer
pdem02$sex <- car::recode(pdem02$demo_sex_v2, "1= 'm';2='f'; else= NA") 
	
#Total Combined Income
#1= Less than $5,000; 2=$5,000 through $11,999; 3=$12,000 through $15,999; 4=$16,000 through $24,999;
#5=$25,000 through $34,999; 6=$35,000 through $49,999; 7=$50,000 through $74,999; 8= $75,000 through $99,999;
#9=$100,000 through $199,999; 10=$200,000 and greater. 999 = Don't know
pdem02$demo_comb_income_v2 <- car::recode(pdem02$demo_comb_income_v2,"c(777,999)=NA")# Combined income

# Caregiver education: demo_prnt_ed_v2
# Partner education:   demo_prtnr_ed_v2
#0 = Never attended/Kindergarten only; 1 = 1st grade; 2 = 2nd grade; 3 = 3rd grade;
#4 = 4th grade; 5 = 5th grade; 6 = 6th grade; 7 = 7th grade; 8 = 8th grade;
#9 = 9th grade; 10 = 10th grade; 11 = 11th grade; 12 = 12th grade; 13 = High school; 14 = GED or equivalent
#15 = Some college; 16 = Associate degree: Occupational; 17 = Associate degree: Academic;
#18 = Bachelor's degree (ex. BA); 19 = Master's degree (ex. MA); 20 = Professional School degree (ex. MD)
#21 = Doctoral degree (ex. PhD); 777 = NA
pdem02$demo_prnt_ed_v2   <- car::recode(pdem02$demo_prnt_ed_v2,"c(777,999)=NA")   #Caregiver education
pdem02$demo_prtnr_ed_v2  <- car::recode(pdem02$demo_prtnr_ed_v2,"c(777,999)=NA") #Partner education
# if partner education is missing, use caregiver education
pdem02$demo_caregiver_ed <- rowMeans(cbind(pdem02$demo_prtnr_ed_v2, pdem02$demo_prnt_ed_v2))
pdem02$demo_caregiver_ed <-  ifelse(is.na(pdem02$demo_caregiver_ed),pdem02$demo_prnt_ed_v2, pdem02$demo_caregiver_ed)

# What race do you consider the child to be
# Recoding to White vs non-white
pdem02$demo_race <- ifelse(
    pdem02 %>% dplyr::select(demo_race_a_p___11:demo_race_a_p___25) %>% rowSums(na.rm = F)==0,0,1
    ) 

# Do you consider the child Hispanic/Latino/Latina
#0='Latinx',1 =non-Latinx
pdem02$demo_ethnicity <- car::recode(pdem02$demo_ethn_v2, "1=0; 2 = 1; 999=NA; 777=NA") 

pdem02$demo_race_ethnicity <- NA
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___10==1] <- 'White'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___11==1] <- 'Black'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___12==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___13==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___14==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___15==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___16==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___17==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___18==1] <- 'Asian'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___19==1] <- 'Asian'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___20==1] <- 'Other'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___21==1] <- 'Asian'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___22==1] <- 'Asian'
pdem02$demo_race_ethnicity[pdem02$demo_race_a_p___23==1] <- 'Asian'
pdem02$demo_race_ethnicity[pdem02$demo_ethn_v2==1] <- "Hispanic"
pdem02$demo_race_ethnicity <- factor(pdem02$demo_race_ethnicity)
pdem02$demo_race_ethnicity <- factor(pdem02$demo_race_ethnicity, 
                                     levels = c("White", "Hispanic", "Black", "Asian", "Other"))
#Other(e.g. Native American, Alaska Native, Pacific Islander)

#demo_prnt_marital_v2
#Are you now married, widowed, divorced, separated, never married or living with a partner? 
# 1 = Married Casado(a) 
# 2 = Widowed Viudo(a)
# 3 = Divorced Divorciado(a)
# 4 = Separated Separado(a)
# 5 = Never married Nunca me he casado 
# 6 = Living with partner
# Recoding for with x without partner
pdem02$demo_prnt_marital <- car::recode(pdem02$demo_prnt_marital_v2,"c(1,6)=1; c(2,3,4,5)=0;777=NA")

#### ABCD ACS Post Stratification Weights ####
acspsw03          <- read.csv(paste0(file_path,"abcd-general/abcd_y_lt.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### Parent Diagnostic Interview (KSADS-5)####
dibf01           <- read.csv(paste0(file_path,"mental-health/mh_p_ksads_bg.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')
dibf01$Adopted_child <-  ifelse(dibf01$kbi_p_c_guard___5==1|dibf01$kbi_p_c_guard___6==1|dibf01$kbi_p_c_guard___7==1,1,0)

#### ABCD Parent Child Behavior Checklist Scores Aseba (CBCL) ####
cbcl             <- read.csv(paste0(file_path,"mental-health/mh_p_cbcl.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### ABCD Parent Family Environment Scale-Family Conflict Subscale Modified from PhenX (FES) ####
fes02            <- read.csv(paste0(file_path,"culture-environment/ce_p_fes.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

fes02$fes_fam_env_score      <-
    fes02 %>% dplyr::select(
        fam_enviro1_p ,
        fam_enviro2r_p ,
        fam_enviro3_p ,
        fam_enviro4r_p ,
        fam_enviro5_p ,
        fam_enviro6_p ,
        fam_enviro7r_p ,
        fam_enviro8_p,
        fam_enviro9r_p
    )  %>% rowSums() # sscep01$fes_p_ss_fc same as fes02$fes_fam_env_score

#### ABCD Sum Scores Culture & Environment Parent ####
# Culture and environment summary scores - parent surveys
sscep01            <- read.csv(paste0(file_path,"culture-environment/ce_p_nsc.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')


# Residential HistoryDerived Scores 
led_l_adi <- read.csv(paste0(file_path,"linked-external-data/led_l_adi.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### ABCD Parent Screen Time Survey (STQ) ####
# Screen time utilization - parent
stq01              <- read.csv(paste0(file_path,"novel-technologies/nt_p_stq.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

stq01$Screentime_week    <- stq01$screentime1_p_hours*60 + stq01$screentime1_p_minutes
stq01$Screentime_weekend <- stq01$screentime2_p_hours*60 + stq01$screentime2_p_minutes
stq01$Average_Screentime <- rowMeans(stq01[,c("Screentime_week","Screentime_weekend")],na.rm = F)/60
stq01$Average_Screentime <- ifelse(
  stq01$Average_Screentime > quantile(stq01$Average_Screentime, probs = c(.995), na.rm = T),
  quantile(stq01$Average_Screentime, probs = c(.995), na.rm = T), stq01$Average_Screentime
  ) # winsorizing outliers to 99.5 percentile

#### ABCD Parent Adult Self Report Scores Aseba (ASR)
asrs01            <- read.csv(paste0(file_path,"mental-health/mh_p_asr.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### ABCD Youth Edinburgh Handedness Inventory Short Form #### abcd_ehis01
abcd_ehis01       <- read.csv(paste0(file_path,"neurocognition/nc_y_ehis.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')
abcd_ehis01$Handedness <- car::recode(abcd_ehis01$ehi_y_ss_scoreb,"1='1.right' ; 2='2.left'; 3='3.mixed'")
  
#### Trauma exposure ####
ptsd01            <- read.csv(paste0(file_path,"mental-health/mh_p_ksads_ptsd.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

ptsd01$Exp_trauma <- ifelse(
    ptsd01 %>% dplyr::select(ksads_ptsd_raw_754_p:ksads_ptsd_raw_770_p) %>% rowSums(na.rm = F) >0, 1, 0
    )

#### 	ABCD Youth NIH TB Summary Scores ####
tbss01            <- read.csv(paste0(file_path,"neurocognition/nc_y_nihtb.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

#### ABCD Developmental History Questionnaire ####
dhx01             <- read.csv(paste0(file_path,"physical-health/ph_p_dhx.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

# Unplanned Pregnancy
dhx01$devhx_6_p <- ifelse(dhx01$devhx_6_p==999,NA, dhx01$devhx_6_p ) # 1 = planned pregnancy
dhx01$devhx_6_p <- car::recode(dhx01$devhx_6_p, "1=0;0=1") # 1 = planned pregnancy #Rec to 1 = unplanned pregnancy
dhx01$devhx_5_p <- ifelse(dhx01$devhx_5_p==999,NA, dhx01$devhx_5_p ) # Does your child have a twin? 1 =yes

# C-section 
dhx01$devhx_13_3_p <- ifelse(dhx01$devhx_13_3_p==999,NA, dhx01$devhx_13_3_p ) #	1 = Yes; 0 = No; 999 = Don't know

# Gestational weeks
# devhx_12a_p	Was the child born prematurely? 1 = Yes ; 0 = No
# devhx_12_p    About how many weeks premature was the child when they were born?
#1 = 1; 2 = 2; 3 = 3; 4 = 4; 5 = 5; 6 = 6; 7 = 7; 8 = 8; 9 = 9; 10 = 10; 11 = 11; 12 = 12;
#13 = Greater than 12/; 999 = Don't know
#example: full term = 40 weeks, born at 36 weeks = 4 weeks premature, born at 32 weeks = 8 weeks premature
#dhx01$devhx_12a_p %>% propfunc()
dhx01$devhx_12a_p <- car::recode(dhx01$devhx_12a_p, "999=NA")

dhx01$GestationalAge <- 40 - ifelse(is.na(dhx01$devhx_12_p) & dhx01$devhx_12a_p == 0,0,  dhx01$devhx_12_p )
dhx01$A_Burden_Gestational_Age   <- ifelse(dhx01$GestationalAge<37, 1,0)
dhx01$A_Burden_Gestational_Age_2   <- ifelse(dhx01$devhx_12_p==999 & dhx01$devhx_12a_p==1,NA,dhx01$A_Burden_Gestational_Age)
# Participants who answered yes to "preterm"(1), but did not provide a number of weeks (999), 
# are currently in the A_Burden_Gestational_Age==1 group. They were dropped from main analysis

dhx01$birth_weight_oz <- ifelse(is.na(dhx01$birth_weight_oz), 0, dhx01$birth_weight_oz)
dhx01$birth_weight_g <- dhx01$birth_weight_lbs *453.592 + dhx01$birth_weight_oz  *28.3495

# Tobacco
dhx01$devhx_8_tobacco <- ifelse(dhx01$devhx_8_tobacco==999,NA, dhx01$devhx_8_tobacco)
#propfunc(dhx01$devhx_8_tobacco)

dhx01$devhx_9_tobacco <- ifelse(dhx01$devhx_9_tobacco==999,NA, dhx01$devhx_9_tobacco)
#propfunc(dhx01$devhx_9_tobacco)

#Alcohol
dhx01$devhx_8_alcohol <- ifelse(dhx01$devhx_8_alcohol==999,NA, dhx01$devhx_8_alcohol)
#dhx01$devhx_8_alcohol %>% propfunc()

dhx01$devhx_9_alcohol <- ifelse(dhx01$devhx_9_alcohol==999,NA, dhx01$devhx_9_alcohol)
#dhx01$devhx_9_alcohol %>% propfunc()

#Marijuana
dhx01$devhx_8_marijuana <- ifelse(dhx01$devhx_8_marijuana==999,NA, dhx01$devhx_8_marijuana)
#dhx01$devhx_8_marijuana%>% propfunc()

dhx01$devhx_9_marijuana <- ifelse(dhx01$devhx_9_marijuana==999,NA, dhx01$devhx_9_marijuana)
#dhx01$devhx_9_marijuana %>% propfunc()

# Pregnancy Complications
# devhx_10a3_p  Severe nausea and vomiting extending past the 6th month or accompanied by weight loss
# devhx_10b3_p  Heavy bleeding requiring bed rest or special treatment
# devhx_10c3_p  Pre-eclampsia, eclampsia, or toxemia? 
# devhx_10d3_p  Severe gall bladder attack? 
# devhx_10e3_p  Persistent proteinuria
# devhx_10f3_p  Rubella during first 3 months of pregnancy
# devhx_10g3_p  Severe anemia
# devhx_10h3_p	Urinary tract infection
# devhx_10i3_p	Pregnancy-related diabetes
# devhx_10j3_p	Pregnancy-related high blood pressure 
# devhx_10k3_p	Previa, abruptio, other problems with placenta
# devhx_10l3_p	Accident or injury requiring medical care
# devhx_10m3_p	Any other conditions requiring medical care

dhx01 <- dhx01 %>%
    dplyr::mutate_at(dplyr::vars(devhx_10a3_p:devhx_10m3_p), ~car::recode(.,"999=NA")) %>%
    dplyr::mutate( A_Burden_Pregnan_Complications = ifelse(rowSums(dplyr::across(devhx_10a3_p:devhx_10m3_p),na.rm = T)> 0,1,0))

# Birth Complications
#devhx_14a3_p Blue at birth
#devhx_14b3_p Slow heart beat
#devhx_14c3_p Did not breathe at first
#devhx_14d3_p Convulsions
#devhx_14e3_p Jaundice needing treatment
#devhx_14f3_p Required oxygen
#devhx_14g_p  Required blood transfusion
#devhx_14h3_p Rh incompatibility 

dhx01 <- dhx01 %>%
    dplyr::mutate_at(dplyr::vars(devhx_14a3_p:devhx_14h3_p), ~car::recode(.,"999=NA")) %>% 
    dplyr::mutate( A_Burden_Birth_Complications = ifelse(rowSums(dplyr::across(devhx_14a3_p:devhx_14h3_p),na.rm = T)>0,1,0))
#dhx01$A_Burden_Birth_Complications %>% propfunc()

#### Birth Weight InterGrowth21
BW_centile <- read.csv(paste0(birthweight_path,"ABCD_birthweight_centiles.csv")) %>%
  filter(!is.na(WeightCentile)) %>%
  dplyr::select(Id, WeightCentile) %>%
  data.table::setnames("Id", "src_subject_id")
BW_centile$A_index_birth_weight <- ifelse(BW_centile$WeightCentile < 9.999, 1, 0)#| BW_centile$WeightCentile > 90.001

#### Genetic Popstrat ####
popstrat <- readxl::read_excel(paste0(gene_path, "ABCD_5_ALL_SUB_projection_pcs.xlsx"))
colnames(popstrat)[1] <- "src_subject_id"

brainvol.gwas <- read.table(paste0(brainvol.gwas.path,"PRS_BV_Jansenetal_2020_ABCD5.all.score"), header = T)
colnames(brainvol.gwas)[c(1,3)] <- c("src_subject_id", "brainvol.pgs")

#### MRI labels ####
mri.labels       <- read.csv("https://nda.nih.gov/api/datadictionary/v2/datastructure/abcd_smrip101/csv")
mri.labels$Type  <-NA
mri.labels$Type[grep("Cortical volume in mm\\^3 of APARC ROI",mri.labels$ElementDescription)]  <- "Cortical volume"
mri.labels$Type[grep("Cortical thickness in mm of APARC ROI",mri.labels$ElementDescription)]   <- "Cortical thickness"
mri.labels$Type[grep("Cortical area in mm\\^2 of APARC ROI",mri.labels$ElementDescription)]    <- "Cortical area"
mri.labels$Type[grep("Cortical sulcal depth in mm of APARC ROI",mri.labels$ElementDescription)]<- "Sulcal depth"

mri.labels$ElementDescription <- gsub("Cortical volume in mm\\^3 of APARC ROI"  ,"Cortical volume",mri.labels$ElementDescription)
mri.labels$ElementDescription <- gsub("Cortical thickness in mm of APARC ROI"   ,"Cortical thickness",mri.labels$ElementDescription)
mri.labels$ElementDescription <- gsub("Cortical area in mm\\^2 of APARC ROI"    ,"Cortical area",mri.labels$ElementDescription)
mri.labels$ElementDescription <- gsub("Cortical sulcal depth in mm of APARC ROI","Sulcal depth",mri.labels$ElementDescription)

mrisubcortical.labels <- read.csv("https://nda.nih.gov/api/datadictionary/v2/datastructure/abcd_smrip201/csv")
# Filter for Subcortical ROIs 
subROIs <- c("smri_vol_scs_ltventriclelh","smri_vol_scs_crbcortexlh","smri_vol_scs_tplh","smri_vol_scs_caudatelh",
  "smri_vol_scs_putamenlh","smri_vol_scs_pallidumlh","smri_vol_scs_hpuslh","smri_vol_scs_amygdalalh",
  "smri_vol_scs_aal","smri_vol_scs_vedclh",
  "smri_vol_scs_ltventriclerh", "smri_vol_scs_crbcortexrh","smri_vol_scs_tprh","smri_vol_scs_caudaterh",
  "smri_vol_scs_putamenrh","smri_vol_scs_pallidumrh", "smri_vol_scs_hpusrh", "smri_vol_scs_amygdalarh",
  "smri_vol_scs_aar","smri_vol_scs_vedcrh")
mrisubcortical.labels <- mrisubcortical.labels[mrisubcortical.labels$ElementName %in% subROIs,]

#### Study Site ID ####
abcd_site <-  read.csv(paste0(file_path,"abcd-general/abcd_y_lt.csv"))%>% dplyr::filter(eventname=='baseline_year_1_arm_1')

```

# Merging datasets

```{r, cache=F}
abcd  <- left_join(mri_info, mri_y_smr_vol_dsk,by='src_subject_id', suffix=c('', '.smr_vol'))
abcd  <- left_join(abcd, mriCortThick,by='src_subject_id', suffix=c('', '.smr_cortThick'))                   
abcd  <- left_join(abcd, mriCortArea, by='src_subject_id', suffix=c('', '.smr_cortArea'))                   
abcd  <- left_join(abcd, mriSulcalDepth, by='src_subject_id', suffix=c('', '.smr_SulcalDepth'))
abcd  <- left_join(abcd, mri_subCortVol_aseg, by='src_subject_id', suffix=c('', '.smr_subVol'))
abcd  <- left_join(abcd, mri_y_qc_incl, by='src_subject_id', suffix=c('', '.qc_incl'))
abcd  <- left_join(abcd, pdem02,   by='src_subject_id', suffix=c('', '.pdem02') )
abcd  <- left_join(abcd, acspsw03, by="src_subject_id", suffix=c('', '.acspsw03')) # Family identification variables
abcd  <- left_join(abcd, dhx01,    by="src_subject_id", suffix=c('', '.dhx01')) 
abcd  <- left_join(abcd, asrs01,   by="src_subject_id", suffix=c('', '.asrs01'))
abcd  <- left_join(abcd, dibf01 %>% select(src_subject_id, Adopted_child), by="src_subject_id")
abcd  <- left_join(abcd, fes02,by="src_subject_id", suffix = c("", ".fes02")) 
abcd  <- left_join(abcd, led_l_adi,by="src_subject_id", suffix = c("", ".adi")) 
abcd  <- left_join(abcd, cbcl, by="src_subject_id", suffix = c("", ".cbcl"))
abcd  <- left_join(abcd, popstrat,by="src_subject_id") 
abcd  <- left_join(abcd, abcd_ehis01 %>% select(src_subject_id, Handedness), by= "src_subject_id" )
abcd  <- left_join(abcd, BW_centile, by= "src_subject_id" )
abcd  <- left_join(abcd,tbss01[,c("src_subject_id","nihtbx_totalcomp_fc",'nihtbx_cryst_fc' ,'nihtbx_fluidcomp_fc')],by= "src_subject_id" )

abcd  <- left_join(abcd, brainvol.gwas[,c("src_subject_id", "brainvol.pgs")],by= "src_subject_id") %>%
  as.data.frame()

abcd$fsqc_qc             <- fsurf_QC$fsqc_qc[match(abcd$src_subject_id, fsurf_QC$src_subject_id)]
abcd$abcd_site           <- abcd_site$site_id_l[match(abcd$src_subject_id, abcd_site$src_subject_id)]
abcd$neighborhood_safety <- sscep01$nsc_p_ss_mean_3_items[match(abcd$src_subject_id, sscep01$src_subject_id)]
abcd$Average_Screentime  <- stq01$Average_Screentime[match(abcd$src_subject_id, stq01$src_subject_id)]
abcd$Exp_trauma          <- ptsd01$Exp_trauma[match(abcd$src_subject_id, ptsd01$src_subject_id)]
abcd$scrn_hr_smoke       <- p_screen$scrn_hr_smoke[match(abcd$src_subject_id, p_screen$src_subject_id)]

# Covariates Labels####
covars.x <- Hmisc::Cs(
    interview_age      , # child‚Äôs age
    demo_comb_income_v2, # Combined income
    demo_caregiver_ed  , # Caregiver education (averaged across both primary caregivers)
    fes_fam_env_score  , # Family Conflict Subscale of the Family Environment Scale;
    devhx_3_p          , # Maternal age at birth
    asr_scr_totprob_t  , # Parent Total Problems
    neighborhood_safety, # reshist_addr1_adi_wsum Area deprivation index / nsc_p_ss_mean_3_items (neigboorhood safety) 
    Average_Screentime,
    PC1,
    PC2,
    PC3,
    PC4,
    PC5,
    PC6,
    PC7,
    PC8,
    PC9,
    PC10,
    brainvol.pgs)

# Computing Prenatal Score
abcd$A_Prenatal_Burden_Score <- abcd %>%
    dplyr::select(devhx_6_p,        # Unplanned pregnancy
           devhx_13_3_p,            # C-section 
           A_Burden_Gestational_Age,
           A_index_birth_weight,
           devhx_8_tobacco,
           devhx_8_alcohol,
           devhx_8_marijuana,
           A_Burden_Pregnan_Complications,
           A_Burden_Birth_Complications) %>% rowSums()

# Filtering Neuroimage QC
abcd      <- abcd[abcd$fsqc_qc ==1 |is.na(abcd$fsqc_qc),] # QC score	0;1	0 = reject; 1 = accept
abcd_full <- abcd

#Split by sex
abcd.f    <- abcd %>% filter(sex=="f")
abcd.m    <- abcd %>% filter(sex=="m")



# Preparing Variables and Labels
# Standardizing Numerical Predictors
for( i in Hmisc::Cs(
    interview_age      , # child‚Äôs age
    demo_comb_income_v2, # Combined income
    demo_caregiver_ed  , # Caregiver education (averaged across both primary caregivers)
    fes_fam_env_score  , # Family Conflict Subscale ofthe Family Environment Scale;
    devhx_3_p          , # Maternal age at birth
    asr_scr_totprob_t  , # Parent Total Problems
    neighborhood_safety,
    Average_Screentime,
    PC1,
    PC2,
    PC3,
    PC4,
    PC5,
    PC6,
    PC7,
    PC8,
    PC9,
    PC10,
    brainvol.pgs,
    smri_vol_scs_intracranialv, 
    smri_thick_cdk_mean,       # Mean Cortical thickness 
    smri_area_cdk_total,       # Total whole brain cortical area in mm^2
    smri_sulc_cdk_mean,         # Mean Sulcal Depth
    A_Prenatal_Burden_Score
)){
  
  abcd.f[,paste0(i,'_z')] <- as.numeric(scale(abcd.f[,i]))
  abcd.m[,paste0(i,'_z')] <- as.numeric(scale(abcd.m[,i]))
}

covars.x <- Hmisc::Cs(
    interview_age_z      , # child‚Äôs age
    demo_comb_income_v2_z, # Combined income
    demo_caregiver_ed_z  , # Caregiver education (averaged across both primary caregivers)
    fes_fam_env_score_z  , # Family Conflict Subscale ofthe Family Environment Scale;
    demo_prnt_marital    , # Presence or absence of a partner for the primary caregiver;
    devhx_3_p_z          , # Maternal age at birth
    asr_scr_totprob_t_z  , # Parent Total Problems
    neighborhood_safety_z,
    Average_Screentime_z,
    Exp_trauma,          #Child‚Äôs exposure to zero vs. 1 significant trauma
    PC1_z,
    PC2_z,
    PC3_z,
    PC4_z,
    PC5_z,
    PC6_z,
    PC7_z,
    PC8_z,
    PC9_z,
    PC10_z,
    brainvol.pgs_z,
    Handedness
    )

covars.x.labels <- data.frame(covars.x=c("A_Prenatal_Burden_Score",covars.x),
                              labels= c("Prenatal Adversity",
                                        "Child‚Äôs age",
                                        "Combined income",
                                        "Caregiver education",
                                        "Family Conflict",
                                        "Relationship status",
                                        "Maternal age",
                                        "Parent Total Problems",
                                        "Neighborhood Safety",
                                        "Screen Time",
                                        "Exposure to Trauma",
                                        "PC1",
                                        "PC2",
                                        "PC3",
                                        "PC4",
                                        "PC5",
                                        "PC6",
                                        "PC7",
                                        "PC8",
                                        "PC9",
                                        "PC10",
                                        "Brain volume genetic score",
                                        "Handedness"
                                        ))


# Filtering Complete cases
abcd.f      <-  abcd.f[complete.cases(abcd.f[,c(covars.x.labels$covars.x, 'smri_vol_scs_intracranialv')]),]
abcd.m      <-  abcd.m[complete.cases(abcd.m[,c(covars.x.labels$covars.x, 'smri_vol_scs_intracranialv')]),]
abcd        <-  rbind(abcd.f, abcd.m)

```

# ABCD Sample Description
A total of `r nrow(abcd)` participants (`r round(prop.table(table(abcd$sex))[2]*100,1)`% males) were considered for the current work based on the availability of demographics, psychosocial, environmental, genetic and of high-quality neuroimaging data (Supplemental eMethods). Ethics approval for the ABCD study was provided by the central Institutional Review Board (IRB) at the University of California, San Diego, and in some instances by local IRB approval. Caregivers provided written informed consent after the procedures had been fully explained, and children assented before participation in the study.
\
\

# Table 1 - Participant characteristics
## Preparing descriptives functions
```{r}
mean.describe2 <- function(x, group){
    
    group.name <- names( table(group))    
    
    g1 <- 
        paste0(
            apply(x[group== group.name[1],]  , 2, mean,na.rm=T) %>% round(2)," (",
            apply(x[group== group.name[1],], 2, sd,na.rm=T) %>% round(2), ")")
    g2 <- 
        paste0(
            apply(x[group== group.name[2],], 2, mean,na.rm=T) %>% round(2)," (",
            apply(x[group== group.name[2],], 2, sd,na.rm=T) %>% round(2), ")")
    
    descritivas <- data.frame(cbind(g1,g2))
    rownames(descritivas) <- names(x)
    colnames(descritivas) <- group.name
    return(descritivas)
}


propfunc <- 
function(x){
    
    require(dplyr)
    
    if(is.null(ncol(x))){
        proportions <- cbind(table(x), round(100*prop.table(table(x)),digits = 2) ) %>% as.data.frame()
        proportions <- cbind(rownames(proportions),proportions)
        proportions <- proportions
        colnames(proportions) <- c("Var", "N", "%")
        return(proportions)
    }else{
    
    propfunc.x <- function(x){
    proportions <- cbind(table(x), round(100*prop.table(table(x)),digits = 2) ) %>% as.data.frame()
    proportions <- cbind(rownames(proportions),proportions)
    colnames(proportions) <- c("Var", "N", "%")
    
    return(proportions)
    }

    x <- apply(x, 2, propfunc.x)
    x <- data.table::rbindlist(x,idcol = T) %>% as.data.frame()
    colnames(x) <- c("Variables", 'Values', "N", "%")
    x$Variables <- ifelse(duplicated(x$Variables), '',x$Variables)
    return(x)}
}

```

## Run descriptive analysis by sex

```{r, warning=FALSE}

num.descriptives <- mean.describe2(x = abcd[,
                        c("A_Prenatal_Burden_Score",
                          'interview_age'      ,# child‚Äôs age
                          'demo_comb_income_v2',# Combined income
                          'demo_caregiver_ed'  ,# Averaged caregivers education
                          'fes_fam_env_score'  ,# Family Conflict Subscale - Family Environment Scale
                          'devhx_3_p'          ,# Maternal age at birth
                          'asr_scr_totprob_t'  ,# Parent Total Problems - ASEBA
                          'reshist_addr1_adi_wsum',
                          'Average_Screentime',
                          "brainvol.pgs_z")],
               group = abcd$sex
                )

rownames(num.descriptives) <- c("Prenatal Adverse Exposure Score, Mean (SD)",
                                "Age in months, Mean (SD)",
                                "Family combined income, Mean (SD)",
                                "Averaged carigiver's education, Mean (SD)",
                                "Family conflict subscale, Mean (SD)",
                                "Maternal age at birth, Mean (SD)",
                                "Parent Total Problems - ASEBA, Mean (SD)",
                                "Neighborhood safety, Mean (SD)",
                                "Average screen time use, Mean (SD)",
                                "Brain volume polygenic score, Mean (SD)"
                                )

colnames(num.descriptives) <- c("Females", 'Males')

# Running t. tests
t.values  <- c()
p.values <- c()
iID <- 0
for( i in c("A_Prenatal_Burden_Score",
            'interview_age'      ,# child‚Äôs age
            'demo_comb_income_v2',# Combined income
            'demo_caregiver_ed'  ,# Averaged caregivers education
            'fes_fam_env_score'  ,# Family Conflict Subscale - Family Environment Scale
            'devhx_3_p'          ,# Maternal age at birth
            'asr_scr_totprob_t'  ,# Parent Total Problems - ASEBA
            'neighborhood_safety',
            'Average_Screentime',
            'brainvol.pgs_z')){
  iID <- iID+1
  p.values[iID] <- t.test(abcd[,i] ~ abcd$sex, var.equal=T)$p.value
  t.values[iID] <- t.test(abcd[,i] ~ abcd$sex, var.equal=T)$statistic
}

num.descriptives$stat.value <- t.values
num.descriptives$p.value <- p.values

prop.descriptives.race              <- by( abcd[,c('demo_race_ethnicity')],abcd$sex,propfunc )
prop.descriptives.race              <- cbind(prop.descriptives.race[[1]],prop.descriptives.race[[2]]) #List to matrix
prop.descriptives.race              <- prop.descriptives.race[,c(2:3,5:6)]# Selecting variables, values and descriptives
prop.descriptives.race$Females      <- paste0(prop.descriptives.race$N, " (", prop.descriptives.race$`%`,")")#
prop.descriptives.race$Males        <- paste0(prop.descriptives.race$N.1, " (", prop.descriptives.race$`%.1`,")")
prop.descriptives.race              <- rbind(NA,prop.descriptives.race)
rownames(prop.descriptives.race)[1] <- "Ethnicity, N (%)"
prop.descriptives.race              <- prop.descriptives.race[,c("Females", 'Males')]
prop.descriptives.race$stat.value   <- NA
prop.descriptives.race$p.value      <- NA
prop.descriptives.race$stat.value[1]<- chisq.test(abcd$demo_race_ethnicity, abcd$sex)$statistic
prop.descriptives.race$p.value[1]   <- chisq.test(abcd$demo_race_ethnicity, abcd$sex)$p.value 


## Preparing Labels
Prenatal.vars <- Hmisc::Cs(devhx_6_p,               # Unplanned pregnancy
                           devhx_13_3_p,            # C-section 
                           A_Burden_Gestational_Age,
                           A_index_birth_weight,
                           devhx_8_tobacco,
                           devhx_8_alcohol,
                           devhx_8_marijuana,
                           A_Burden_Pregnan_Complications,
                           A_Burden_Birth_Complications)

prop.descriptives.PrenatalAdversity              <- by( abcd[,Prenatal.vars],abcd$sex,propfunc )
prop.descriptives.PrenatalAdversity              <- cbind(prop.descriptives.PrenatalAdversity[[1]],prop.descriptives.PrenatalAdversity[[2]]) #List to matrix
prop.descriptives.PrenatalAdversity              <- prop.descriptives.PrenatalAdversity[seq(2,18, by=2),c(3:4,7:8)]# Selecting variables, values and descriptives
prop.descriptives.PrenatalAdversity$Females      <- paste0(prop.descriptives.PrenatalAdversity$N, " (", prop.descriptives.PrenatalAdversity$`%`,")")#
prop.descriptives.PrenatalAdversity$Males        <- paste0(prop.descriptives.PrenatalAdversity$N.1, " (", prop.descriptives.PrenatalAdversity$`%.1`,")")
prop.descriptives.PrenatalAdversity              <- rbind(NA,prop.descriptives.PrenatalAdversity)
rownames(prop.descriptives.PrenatalAdversity)[1] <- "Adverse Prenatal Exposures, N (%)"
prop.descriptives.PrenatalAdversity              <- prop.descriptives.PrenatalAdversity[,c("Females", 'Males')]
prop.descriptives.PrenatalAdversity$stat.value   <- NA
prop.descriptives.PrenatalAdversity$p.value      <- NA

iID <- 1
for( i in Prenatal.vars){
    iID <- iID+1
    prop.descriptives.PrenatalAdversity$stat.value[iID]<- chisq.test(abcd[,i], abcd$sex)$statistic
    prop.descriptives.PrenatalAdversity$p.value[iID]   <- chisq.test(abcd[,i], abcd$sex)$p.value     
    
}

Prenatal.labels <- c("Unplanned pregnancy",
                    "C-Section",
                    "Gestational Age",
                    "Birth weight",
                    "Tobacco use",
                    "Alcohol use",
                    "Marijuana use",
                    "Pregnancy complications",
                    "Birth complications")

rownames(prop.descriptives.PrenatalAdversity) <- c("Adverse Prenatal Exposures, N (%)", Prenatal.labels )

# Merging Descriptives
participants.descriptives      <- rbind(num.descriptives,prop.descriptives.race,prop.descriptives.PrenatalAdversity)

```


## Create Table 1
```{r}

kable(participants.descriptives, digits = 3, caption = "Participants characteristics") %>%
  kable_styling(position = "center")

```

# Figures 1 and 2
## Define LME function

```{r LME function, cache=F, warning=FALSE, message=FALSE}

LME_loop <-
  function(predictors,
           outcomes,
           covars = NULL,
           data,
           predictorsLab = NULL,
           outcomesLab = NULL,
           ROI=NULL,
           print.model = NULL,
           dig = NULL,
           Scale=NULL,
           is.log10 = FALSE
  ){
    require(dplyr)
    
    interac_names   <- c("MainEffect_ID",
                         "Outcome",
                         "Label",
                         'ElementDescription',
                         "Predictor",
                         "Beta_Predictor",
                         'Std.error',
                         "Sample_size",
                         "R2",
                         "p_value_Predictor"
                         )
    
    
    #Labels
    
    if(!is.null(predictorsLab)){
        
        predictors <-  predictors
        predictorsLabels <- data.frame(vars = predictors,
                                       names= predictorsLab
                                       ,stringsAsFactors = F)
    }
    
    if(!is.null(outcomesLab)){
        
        outcomes <-  outcomes
        outcomesLabels <- data.frame(vars = outcomes,
                                     names= outcomesLab,
                                     ROI=ROI,
                                     stringsAsFactors = F)
    }
    #Results 
    main_results <- matrix(NA, nrow = length(predictors)*length(outcomes), ncol = 10) %>% as.data.frame()
    colnames(main_results) <- interac_names
    
    #Preparing data
    
    if (is.null(covars)) {
        lm.formula <- "outcome.y ~ predictor.x"
    } else{
        lm.formula <-
            paste("outcome.y ~ predictor.x + ", paste(covars, collapse = "+"))
    }
    if(is.log10==TRUE){lm.formula <- gsub("outcome.y", "log10(outcome.y)", lm.formula)}
    if(isTRUE(Scale))   {lm.formula <- gsub("outcome.y", "scale(outcome.y)", lm.formula)}
    
    
    iID <- 0
    for(i in outcomes){
        for(j in predictors){
            
            iID <- iID+1
            data$outcome.y    <- data[,i]  
            data$predictor.x  <- data[,j]
            
            if (is.null(covars)) {
              data1 <- data[ complete.cases( data[,c("outcome.y", "predictor.x")] ), ]#covars
            } else{
              data1 <- data[ complete.cases( data[,c("outcome.y", "predictor.x", covars)] ), ]#covars
            }
            
            main_results$Sample_size   [iID] <- nrow(data1)
            main_results$MainEffect_ID [iID] <- iID
            main_results$Outcome       [iID] <- i
            
            if(!is.null(outcomesLab)){
                main_results$Label              [iID] <- outcomesLabels[outcomesLabels$vars== i,"names"]
                main_results$ElementDescription [iID] <- outcomesLabels[outcomesLabels$vars== i,"ROI"]
            }
            
            if(!is.null(predictorsLab)){
                main_results$Predictor [iID] <- predictorsLabels[predictorsLabels$vars== j,"names"]
            }else{main_results$Predictor  [iID] <- j}
            
            
            if(nrow(data1)<15)next
            
            formula.random="~(1|mri_info_deviceserialnumber/rel_family_id)"
            
            model.loop = gamm4::gamm4(
              as.formula(lm.formula),
              random = as.formula(formula.random),
              weights = NULL,
              data = data1
            )

            main_results$Beta_Predictor      [iID] <- summary(model.loop$gam)$p.table[2,1] 
            main_results$Std.error           [iID] <- summary(model.loop$gam)$p.table[2,2] 
            main_results$p_value_Predictor   [iID] <- summary(model.loop$gam)$p.table[2,4] 
            main_results$R2                  [iID] <- summary(model.loop$gam)$r.sq %>% round(3)
            
            if (!is.null(print.model)) {
                print(i)
                print(j)
                print(summary(model.loop$gam)$p.table)
        }
        
    }
    
    }
    
    main_results <<- main_results
  }

```

# Figure 1
# Cortical Volume - Obtaining FDR-significant Betas for plotting
## Running analysis

```{r Cortical Volume, cache=TRUE, warning=FALSE, message=FALSE}
CVol_outcomes           <- mri.labels[mri.labels$Type %in% "Cortical volume", "ElementName"]
CvolROIs                <- mri.labels[mri.labels$Type %in% "Cortical volume", "ElementDescription"]
CvolMRI_outcomes_Labels <- mri.labels[mri.labels$Type %in% "Cortical volume","Aliases"]

CortVolResults.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CVol_outcomes,
    outcomesLab = CvolMRI_outcomes_Labels,
    ROI = CvolROIs,
    Scale = T,
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    data = abcd.f
  )
CortVolResults.F$FDR <- p.adjust(CortVolResults.F$p_value_Predictor,"fdr")

CortVolResults.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CVol_outcomes,
    outcomesLab = CvolMRI_outcomes_Labels,
    ROI = CvolROIs,
    Scale = T,
    covars = c(covars.x,'smri_vol_scs_intracranialv_z'),
    data = abcd.m
  )

CortVolResults.M$FDR <- p.adjust(CortVolResults.M$p_value_Predictor,"fdr")

```

## Cortical Volume Girls

```{r}
kable(CortVolResults.F[CortVolResults.F$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Cortical Volume Significant Regions - Females',
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

## Cortical Volume Boys

```{r}
kable(CortVolResults.M[CortVolResults.M$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = "Cortical Volume Significant Regions - Males",
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

# Subcortical Volume - Obtaining FDR-significant Betas for plotting
## Running analysis

```{r SubCortical, cache=TRUE}
SubCortical_outcomes        <- mrisubcortical.labels$ElementName
SubCortical_outcomes_Labels <- mrisubcortical.labels$Aliases
SubROIs                     <- mrisubcortical.labels$ElementDescription
SubROIs                     <- gsub("Volume in mm\\^3 of ASEG ROI ", "Subcortical volume ", SubROIs)

SubCortVolResults.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  SubCortical_outcomes,
    outcomesLab = SubCortical_outcomes_Labels,
    ROI = SubROIs,
    Scale = T,
    covars = c(covars.x, 'smri_vol_scs_intracranialv_z'),
    data = abcd.f
  )
SubCortVolResults.F$FDR <- p.adjust(SubCortVolResults.F$p_value_Predictor,"fdr")

SubCortVolResults.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score",
    outcomes =  SubCortical_outcomes,
    outcomesLab = SubCortical_outcomes_Labels,
    ROI = SubROIs,
    Scale = T,
    covars = c(covars.x, 'smri_vol_scs_intracranialv_z'),
    data = abcd.m
  )

SubCortVolResults.M$FDR <- p.adjust(SubCortVolResults.M$p_value_Predictor,"fdr")

```

## Subcortical Volume Girls

```{r}
kable(SubCortVolResults.F[SubCortVolResults.F$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = "Subcortical Volume Significant Regions - Females", 
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

## Subcortical Volume Boys

```{r}
kable(SubCortVolResults.M[SubCortVolResults.M$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Subcortical Volume Significant Regions - Males',
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

# Cortical Thickness - Obtaining FDR-significant Betas for plotting
## Running analysis

```{r Cortical Thickness, cache=TRUE}
CortThick_outcomes      <- mri.labels[mri.labels$Type %in% "Cortical thickness", "ElementName"]
CortThickROIs           <- mri.labels[mri.labels$Type %in% "Cortical thickness", "ElementDescription"]
CortThick_outcomes_Labels <- mri.labels[mri.labels$Type %in% "Cortical thickness","Aliases"]

CortThickResults.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CortThick_outcomes,
    outcomesLab = CortThick_outcomes_Labels,
    ROI = CortThickROIs,
    Scale = T,
    covars = c(covars.x, 'smri_thick_cdk_mean_z'),
    data = abcd.f
  )
CortThickResults.F$FDR <- p.adjust(CortThickResults.F$p_value_Predictor,"fdr")

CortThickResults.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CortThick_outcomes,
    outcomesLab = CortThick_outcomes_Labels,
    ROI = CortThickROIs,
    covars = c(covars.x, 'smri_thick_cdk_mean_z'),
    Scale = T,
    data = abcd.m
  )

CortThickResults.M$FDR <- p.adjust(CortThickResults.M$p_value_Predictor,"fdr")
```

## Cortical Thickness Girls

```{r}
kable(CortThickResults.F[CortThickResults.F$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Cortical Thickness Significant Regions - Females',
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

## Cortical Thickness Boys

```{r}
kable(CortThickResults.M[CortThickResults.M$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Cortical Thickness Significant Regions - Males',
      digits = 3,
      row.names = F) %>%
  kableExtra::kable_styling(position = "center")
```
# Figure 2
# Cortical Area - Obtaining FDR-significant Betas for plotting
## Running analysis

```{r CorticalArea , cache=TRUE}
CortArea_outcomes      <- mri.labels[mri.labels$Type %in% "Cortical area", "ElementName"]
CortAreaROIs           <- mri.labels[mri.labels$Type %in% "Cortical area", "ElementDescription"]
CortArea_outcomes_Labels <- mri.labels[mri.labels$Type %in% "Cortical area","Aliases"]

CortAreaResults.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CortArea_outcomes,
    outcomesLab = CortArea_outcomes_Labels,
    covars = c(covars.x,'smri_area_cdk_total_z'),
    ROI = CortAreaROIs,
    Scale = T,
    data = abcd.f
  )

CortAreaResults.F$FDR <- p.adjust(CortAreaResults.F$p_value_Predictor,"fdr")

CortAreaResults.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  CortArea_outcomes,
    outcomesLab = CortArea_outcomes_Labels,
    ROI = CortAreaROIs,
    Scale = T,
    covars = c(covars.x,'smri_area_cdk_total_z'),
    data = abcd.m
  )

CortAreaResults.M$FDR <- p.adjust(CortAreaResults.M$p_value_Predictor,"fdr")

```

## Cortical Area Girls

```{r}
kable(CortAreaResults.F[CortAreaResults.F$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Cortical Area Significant Regions - Females',
      digits = 3, 
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```

## Cortical Area Boys

```{r}
kable(CortAreaResults.M[CortAreaResults.M$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = 'Cortical Area Significant Regions - Males',
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")

```

# Sulcal Depth  - Obtaining FDR-significant Betas for plotting
## Running analysis

```{r Sulcal depth,cache=TRUE}

SulcalDepth_outcomes      <- mri.labels[mri.labels$Type %in% "Sulcal depth", "ElementName"]
SulcalDepthROIs           <- mri.labels[mri.labels$Type %in% "Sulcal depth", "ElementDescription"]
SulcalDepth_outcomes_Labels <- mri.labels[mri.labels$Type %in% "Sulcal depth","Aliases"]

SulcalDepthResults.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  SulcalDepth_outcomes,
    outcomesLab = SulcalDepth_outcomes_Labels,
    ROI = SulcalDepthROIs,
    Scale = T,
    covars = c(covars.x,'smri_sulc_cdk_mean_z'),
    data = abcd.f
  )
SulcalDepthResults.F$FDR <- p.adjust(SulcalDepthResults.F$p_value_Predictor,"fdr")

SulcalDepthResults.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  SulcalDepth_outcomes,
    outcomesLab = SulcalDepth_outcomes_Labels,
    covars = c(covars.x,'smri_sulc_cdk_mean_z'),
    ROI = SulcalDepthROIs,
    Scale = T,
    data = abcd.m
  )

SulcalDepthResults.M$FDR <- p.adjust(SulcalDepthResults.M$p_value_Predictor,"fdr")

```

## Sulcal Depth Girls

```{r}
kable(SulcalDepthResults.F[SulcalDepthResults.F$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = "Sulcal Depth Significant Regions - Females",
      digits = 3,
      row.names = F) %>%
  kableExtra::kable_styling(position = "center")

```

## Sulcal Depth Boys

```{r}
kable(SulcalDepthResults.M[SulcalDepthResults.M$FDR<0.05,-c(1, 3)],#Omit iID and Aliases
      caption = "Sulcal Depth Siginificant Regions - Males",
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")

```


# Figure 3
# Figure 3A
## Select significant brain region measures

```{r}
#Selecting Significant ROIs
Main_results.F <- rbind(CortVolResults.F,   
                        SubCortVolResults.F,
                        CortThickResults.F,
                        CortAreaResults.F,  
                        SulcalDepthResults.F)

Main_results.M <- rbind(CortVolResults.M,
                        SubCortVolResults.M,
                        CortThickResults.M,
                        CortAreaResults.M,
                        SulcalDepthResults.M)

```

## Factor loadings
```{r }
# N significant Brain Regions
SigROIsN.f <- sum(Main_results.F$FDR<0.05)
SigROIsN.m <- sum(Main_results.M$FDR<0.05)

# Running Principal component analysis ####
# Females
PCs.F                                <- principal(abcd.f[,Main_results.F$Outcome[Main_results.F$FDR<0.05]], 3)
BrainPCsLoadings.F                   <- as.data.frame(PCs.F$loadings[1:SigROIsN.f, 1:3])
rownames(BrainPCsLoadings.F)         <- Main_results.F$ElementDescription [Main_results.F$FDR<0.05]
for(i in 1:3){BrainPCsLoadings.F[,i] <- ifelse(abs(BrainPCsLoadings.F[,i])<.2,0, BrainPCsLoadings.F[,i])}

BrainPCsLoadings.F                   <- arrange(BrainPCsLoadings.F, desc(RC3))
BrainPCsLoadings.F                   <- arrange(BrainPCsLoadings.F, desc(RC2))
BrainPCsLoadings.F                   <- arrange(BrainPCsLoadings.F, desc(RC1))
BrainPCsLoadings.F$ElementDescription<- Main_results.F$ElementDescription [Main_results.F$FDR<0.05]

# Re-ordering Cols
colnames(BrainPCsLoadings.F)[1:3]    <- c('Brain PC1', 'Brain PC2','Brain PC3')#renaming PCs
BrainPCsLoadings.F.long              <- reshape2::melt(BrainPCsLoadings.F,
                                             id="ElementDescription",
                                             measure=c('Brain PC1', 'Brain PC2','Brain PC3'),
                                             variable.name="Brain components", value.name="Loading")

BrainPCsLoadings.F.long$ElementDescription <- factor(BrainPCsLoadings.F.long$ElementDescription,
                                                     levels = BrainPCsLoadings.F$ElementDescription)

#Males
PCs.M                                <- principal(abcd.m[,Main_results.M$Outcome[Main_results.M$FDR<0.05]], 3)
BrainPCsLoadings.M                   <- as.data.frame(PCs.M$loadings[1:SigROIsN.m, 1:3])
rownames(BrainPCsLoadings.M)         <- Main_results.M$ElementDescription [Main_results.M$FDR<0.05]
for(i in 1:3){BrainPCsLoadings.M[,i] <- ifelse(abs(BrainPCsLoadings.M[,i])<.2,0, BrainPCsLoadings.M[,i])}

BrainPCsLoadings.M                   <- arrange(BrainPCsLoadings.M, desc(RC3))
BrainPCsLoadings.M                   <- arrange(BrainPCsLoadings.M, desc(RC2))
BrainPCsLoadings.M                   <- arrange(BrainPCsLoadings.M, desc(RC1))
BrainPCsLoadings.M$ElementDescription<- Main_results.M$ElementDescription [Main_results.M$FDR<0.05]

# Re-ordering Cols
colnames(BrainPCsLoadings.M)[1:3]    <- c('Brain PC1', 'Brain PC2','Brain PC3')#renaming PCs
BrainPCsLoadings.M.long              <- reshape2::melt(BrainPCsLoadings.M,
                                             id="ElementDescription",
                                             measure=c('Brain PC1', 'Brain PC2','Brain PC3'),
                                             variable.name="Brain components", value.name="Loading")

BrainPCsLoadings.M.long$ElementDescription <- factor(BrainPCsLoadings.M.long$ElementDescription,
                                                     levels = BrainPCsLoadings.M$ElementDescription)

```

## Create Figure 3A - Factor lodings significant brain regions

```{r, fig.height=6, fig.width=12}

Loadings.F <-
  ggplot(BrainPCsLoadings.F.long,aes(y = ElementDescription,x = round(Loading,1),abs(Loading),fill = Loading)) +
  facet_wrap( ~ `Brain components`, nrow = 1) +
  geom_bar(stat = "identity") +
  scale_fill_gradient2(name = "Loading", high = "red", mid = "white", low = "blue", midpoint = 0, guide = 'none') +
  theme_bw(base_size = 10)+
  geom_vline(xintercept = 0, linewidth=.25)+
  coord_cartesian(xlim = c(-.6,.8))+
    labs(y="Brain Data Modalities and Regions", title = 'Females')+
  theme(axis.text.y = element_text(size = 8),
        axis.text.x = element_text(size = 8),
        plot.title = element_text(size = 11,hjust = 0.5),
        axis.title.x =  element_blank(),
        text=element_text(family="Arial"))

Loadings.M <-
  ggplot(BrainPCsLoadings.M.long,aes(y = ElementDescription,x = round(Loading,1),abs(Loading),fill = Loading)) +
  facet_wrap( ~ `Brain components`, nrow = 1) +
  geom_bar(stat = "identity") +
  scale_fill_gradient2(name = "Loading", high = "red", mid = "white", low = "blue", midpoint = 0, guide = 'none') +
  coord_cartesian(xlim = c(-.6,.8))+
  theme_bw(base_size = 10) +
  geom_vline(xintercept = 0, linewidth=.25)+
  labs(y="Brain Data Modalities and Regions", title = 'Males')+
  theme(axis.text.y = element_text(size = 8),
        axis.text.x = element_text(size = 8),
        plot.title = element_text(size = 11,hjust = 0.5),
        axis.title.x =  element_blank(),
        text=element_text(family="Arial"))

grid.arrange(Loadings.F, Loadings.M, ncol=2)
jpeg(filename = "Figure3A.jpg", height =4*1.25 ,width = 8*1.5, units = 'in', res = 600)
grid.arrange(Loadings.F, Loadings.M, ncol=2)
dev.off()

```

# Figure 3B
## Analysis Brain PCs and children's behavior and cognition

```{r cbcl and brain PCs, cache=TRUE}

# Exploring PCs and CBCL and Cognitive measures

abcd.f[,'Brain.PC1'] <-  as.numeric(PCs.F$scores[,1])
abcd.f[,'Brain.PC2'] <-  as.numeric(PCs.F$scores[,2])
abcd.f[,'Brain.PC3'] <-  as.numeric(PCs.F$scores[,3])

abcd.m[,'Brain.PC1'] <-  as.numeric(PCs.M$scores[,1])
abcd.m[,'Brain.PC2'] <-  as.numeric(PCs.M$scores[,2])
abcd.m[,'Brain.PC3'] <-  as.numeric(PCs.M$scores[,3])

CBCL_results.F <- LME_loop(
  predictors = c('Brain.PC1', 'Brain.PC2', "Brain.PC3"),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.f
)

CBCL_results.F$FDR <- p.adjust(CBCL_results.F$p_value_Predictor, 'fdr')

CBCL_results.M <- LME_loop(
  predictors = c('Brain.PC1', 'Brain.PC2', "Brain.PC3"),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.m
)

CBCL_results.M$FDR <- p.adjust(CBCL_results.M$p_value_Predictor, 'fdr')

```

## Associations Between Brain Components with Behavior Problems and Cognition
## Create eTable3

```{r, CBCL_results.F}

kable(cbind(CBCL_results.F[,-c(1,3:4)],CBCL_results.M[,-c(1:5)]), 
      digits = 4, 
      caption = "eTable 3. Associations Brain Components with Behavior Problems and Cognition",row.names = F)%>%
  kableExtra::kable_styling(position = "center")%>%
  add_header_above(c(" "=2, 'Females'=6, 'Males'=6))

```

## Associations Between Prenatal Adversity with Children's Behavior and Cognition

```{r}
Prenatal_results.F <- LME_loop(
  predictors = c('A_Prenatal_Burden_Score_z'),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.f
)

Prenatal_results.F$FDR <- p.adjust(Prenatal_results.F$p_value_Predictor, 'fdr')

Prenatal_results.M <- LME_loop(
  predictors = c('A_Prenatal_Burden_Score'),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.m
)

Prenatal_results.M$FDR <- p.adjust(Prenatal_results.M$p_value_Predictor, 'fdr')

```

## Create eTable4
```{r}
kable(cbind(Prenatal_results.F[,-c(1,3:4)],Prenatal_results.M[,-c(1:5)]), 
      digits = 3, 
      caption = "eTable4. Associations Prenatal Adverse Exposures with Behavior Problems and Cognition",row.names = F)%>%
  kableExtra::kable_styling(position = "center")%>%add_header_above(c(" "=2, 'Females'=6, 'Males'=6))

```


## Specifing mediation models

```{r Mediation, cache=TRUE}

# Standardizing predictors for mediation ####
for( i in c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
)){
  
  abcd.f[,paste0(i,'.std')] <- as.numeric(scale(abcd.f[,i]))
  abcd.m[,paste0(i,'.std')] <- as.numeric(scale(abcd.m[,i]))
  }

#### Specifing Mediation Models ####
# Mediation Brain PC1 ####

#Total problems
med.BrainPC1.Tot <- "
    # direct effect
    cbcl_scr_syn_totprob_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_totprob_t.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_totprob_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

#Externalizing problems - PC1
med.BrainPC1.Ext <- "
    # direct effect
    cbcl_scr_syn_external_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_external_t.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2 ~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_external_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"
# Internalizing
med.BrainPC1.Int <- "
    # direct effect
    cbcl_scr_syn_internal_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_internal_t.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_internal_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

# Total Cognition
med.BrainPC1.CogTotal <- "
    # direct effect
    nihtbx_totalcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_totalcomp_fc.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_totalcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.BrainPC2.CogTotal <-  "
    # direct effect
    nihtbx_totalcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC2 ~ a*A_Prenatal_Burden_Score
    nihtbx_totalcomp_fc.std ~ b*Brain.PC2
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_totalcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"


# Fluid Cognition - PC1 and PC2
med.BrainPC1.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.BrainPC2.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC2 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*Brain.PC2
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.BrainPC3.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    
    Brain.PC3 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*Brain.PC3
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2 ~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

# Cryst Cognition
med.BrainPC1.CogCryst <- "
    # direct effect
    nihtbx_cryst_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    Brain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_cryst_fc.std ~ b*Brain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    Brain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    Brain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_cryst_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

#Mediation Females

fit.BrainPC1.CogTotal.f <- lavaan::sem(med.BrainPC1.CogTotal, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.BrainPC1.CogFluid.f <- lavaan::sem(med.BrainPC1.CogFluid, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.BrainPC1.CogCryst.f <- lavaan::sem(med.BrainPC1.CogCryst, data = abcd.f, se = "bootstrap", bootstrap = 1000)

fit.BrainPC1.Ext.f      <- lavaan::sem(med.BrainPC1.Ext, data = abcd.f, se = "bootstrap", bootstrap = 1000)

fit.BrainPC2.CogFluid.f <- lavaan::sem(med.BrainPC2.CogFluid, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.BrainPC2.CogTotal.f <- lavaan::sem(med.BrainPC2.CogTotal, data = abcd.f, se = "bootstrap", bootstrap = 1000)

#Mediation Males
fit.BrainPC1.Tot.m      <- lavaan::sem(med.BrainPC1.Tot     , data = abcd.m, se = "bootstrap", bootstrap = 1000)
fit.BrainPC1.Ext.m      <- lavaan::sem(med.BrainPC1.Ext     , data = abcd.m, se = "bootstrap", bootstrap = 1000)


```
## Preparing Mediation Results
```{r}
#Females
fit.Pars.BrainPC1.CogTotal.f <- lavaan::parameterestimates(fit.BrainPC1.CogTotal.f, boot.ci.type = "bca.simple") 
fit.Pars.BrainPC1.CogFluid.f <- lavaan::parameterestimates(fit.BrainPC1.CogFluid.f, boot.ci.type = "bca.simple") 
fit.Pars.BrainPC1.CogCryst.f <- lavaan::parameterestimates(fit.BrainPC1.CogCryst.f, boot.ci.type = "bca.simple") 

fit.Pars.PC1.Ext.f           <- lavaan::parameterestimates(fit.BrainPC1.Ext.f, boot.ci.type = "bca.simple") 

fit.Pars.BrainPC2.CogFluid.f <- lavaan::parameterestimates(fit.BrainPC2.CogFluid.f, boot.ci.type = "bca.simple") 
fit.Pars.BrainPC2.CogTotal.f <- lavaan::parameterestimates(fit.BrainPC2.CogTotal.f, boot.ci.type = "bca.simple") 

# Males
fit.Pars.BrainPC1.Tot.m      <- lavaan::parameterestimates(fit.BrainPC1.Tot.m, boot.ci.type = "bca.simple") 
fit.Pars.BrainPC1.Ext.m      <- lavaan::parameterestimates(fit.BrainPC1.Ext.m, boot.ci.type = "bca.simple") 

```

## Create Figure 3B - Mediation analysis parameters used in Figure 3B

## Mediation Parameters Females
```{r}
kable(fit.Pars.BrainPC1.CogTotal.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Total Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.BrainPC1.CogFluid.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.BrainPC1.CogCryst.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")


kable(fit.Pars.PC1.Ext.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Externalizing Problems  - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.BrainPC2.CogFluid.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC2 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.BrainPC2.CogTotal.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC2 for Total Cognition - Females") %>% 
  kable_styling(position = "center")


```

## Mediation Parameters Males


```{r}

kable(fit.Pars.BrainPC1.Tot.m %>% filter(label!=""), digits = 3,caption = "Mediation parameters for Total Problems - Males") %>% 
  kable_styling(position = "center")

kable(fit.Pars.BrainPC1.Ext.m %>% filter(label!=""), digits = 3,caption = "Mediation parameters for Externalizing Problems - Males") %>% 
  kable_styling(position = "center")

```


# Figure 4 - Meta-analysis of significant Findings
## Define LME By scanner Function

```{r}
LME_bySite_loop <-
  function(predictors,
           outcomes,
           covars = NULL,
           data,
           predictorsLab = NULL,
           outcomesLab = NULL,
           ROI=NULL,
           print.model = NULL,
           dig = NULL,
           is.log10 = FALSE,
           Site
  ){
    require(dplyr)
    
    interac_names   <- c("MainEffect_ID",
                         "Site",
                         "Outcome",
                         "Label",
                         'ElementDescription',
                         "Predictor",
                         "Beta_Predictor",
                         'Std.error',
                         "Sample_size",
                         "R2",
                         "p_value_Predictor"
                         )
    
    
    #Labels
    
    if(!is.null(predictorsLab)){
        
        predictors       <-  predictors
        predictorsLabels <- data.frame(vars = predictors,
                                       names= predictorsLab
                                       ,stringsAsFactors = F)
    }
    
    if(!is.null(outcomesLab)){
      
      outcomes <-  outcomes
      outcomesLabels <- data.frame(vars = outcomes,
                                   names= outcomesLab,
                                   ROI=ROI,
                                   stringsAsFactors = F)
      }
    
    #Results 
    site.name <-  names(table(data[,Site]))
    main_results <- matrix(NA, nrow = length(predictors)*length(outcomes)*length(site.name), ncol = 11) %>% as.data.frame()
    colnames(main_results) <- interac_names
    
    #Preparing data
    
    if (is.null(covars)) {
        lm.formula <- "outcome.y ~ predictor.x"
    } else{
        lm.formula <-
            paste("scale(outcome.y) ~ predictor.x + ", paste(covars, collapse = "+"))
    }
    if(is.log10==TRUE){lm.formula <- gsub("outcome.y", "log10(outcome.y)", lm.formula)}
    
    formula.random = "~(1|rel_family_id)"
    
    iID <- 0
    for(i in outcomes){
      for(j in predictors){
          try(silent = T,
        for(k in site.name){
          
            Data_Site <- data[data[,Site] == k,]
          
            iID <- iID+1
            Data_Site$outcome.y    <- Data_Site[,i]  
            Data_Site$predictor.x  <- Data_Site[,j]
            
            if(is.null(covars)){
              Data_Site1 <- Data_Site[ complete.cases( Data_Site[,c("outcome.y", "predictor.x")] ), ]
            }else{
              Data_Site1 <- Data_Site[ complete.cases( Data_Site[,c("outcome.y", "predictor.x", covars)] ), ]
              }
            
            main_results$Sample_size   [iID] <- nrow(Data_Site1)
            main_results$MainEffect_ID [iID] <- iID
            main_results$Site          [iID] <- k
            main_results$Outcome  [iID] <- i
            
            
            if(!is.null(outcomesLab)){
                main_results$Label              [iID] <- outcomesLabels[outcomesLabels$vars== i,"names"]
                main_results$ElementDescription [iID] <- outcomesLabels[outcomesLabels$vars== i,"ROI"]
            }
            
            
            if(!is.null(predictorsLab)){
                main_results$Predictor [iID] <- predictorsLabels[predictorsLabels$vars== j,"names"]
            }else{main_results$Predictor  [iID] <- j}
            
            
            if(nrow(Data_Site)<49)next
            
            
            #Running Model
            model.loop = gamm4::gamm4(
              as.formula(lm.formula),
              random = as.formula(formula.random),
              weights = NULL,
              data = Data_Site
            )

            main_results$Beta_Predictor      [iID] <- summary(model.loop$gam)$p.table[2,1] 
            main_results$Std.error           [iID] <- summary(model.loop$gam)$p.table[2,2] 
            main_results$p_value_Predictor   [iID] <- summary(model.loop$gam)$p.table[2,4] 
            main_results$R2                  [iID] <- summary(model.loop$gam)$r.sq %>% round(3)
            
            
            if (!is.null(print.model)) {
                print(i)
                print(j)
                print(k)
                print(summary(model.loop$gam)$p.table)
        
        }
    }
          )    
    
    }
    }
    main_results <<- main_results
}

```



## Analysis by scanner for meta-analysis

```{r Analysis by site, message=FALSE, warning=FALSE, error=FALSE, eval=TRUE, cache=TRUE }

#Cortical Volume by site Girls
CortVol_SigResults_bySite.F <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortVolResults.F$Outcome            [CortVolResults.F$FDR <0.05],
    outcomesLab   =  CortVolResults.F$Label              [CortVolResults.F$FDR <0.05],
    ROI           =  CortVolResults.F$ElementDescription [CortVolResults.F$FDR <0.05],
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.f
  )

CortVol_SigResults_bySite.F      <- 
  CortVol_SigResults_bySite.F[complete.cases(CortVol_SigResults_bySite.F),]#filtering GLMEs that didnt fit 

CortVol_SigResults_bySite.F$Site <- 
  abcd$abcd_site[match(CortVol_SigResults_bySite.F$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Cortical Volume by site Boys
CortVol_SigResults_bySite.M <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortVolResults.M$Outcome            [CortVolResults.M$FDR <0.05],
    outcomesLab   =  CortVolResults.M$Label              [CortVolResults.M$FDR <0.05],
    ROI           =  CortVolResults.M$ElementDescription [CortVolResults.M$FDR <0.05],
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.m
  )

CortVol_SigResults_bySite.M      <- 
  CortVol_SigResults_bySite.M[complete.cases(CortVol_SigResults_bySite.M),]#filtering GLMEs that didnt fit 

CortVol_SigResults_bySite.M$Site <- 
  abcd$abcd_site[match(CortVol_SigResults_bySite.M$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#SubCortical Volume by site Girls
SubCortVol_SigResults_bySite.F <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  SubCortVolResults.F$Outcome            [SubCortVolResults.F$FDR <0.05],
    outcomesLab   =  SubCortVolResults.F$Label              [SubCortVolResults.F$FDR <0.05],
    ROI           =  SubCortVolResults.F$ElementDescription [SubCortVolResults.F$FDR <0.05],
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.f
  )

SubCortVol_SigResults_bySite.F      <- 
  SubCortVol_SigResults_bySite.F[complete.cases(SubCortVol_SigResults_bySite.F),]#filtering GLMEs that didnt fit 

SubCortVol_SigResults_bySite.F$Site <- 
  abcd$abcd_site[match(SubCortVol_SigResults_bySite.F$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI


#SubCortical Volume by site Boys
SubCortVol_SigResults_bySite.M <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  SubCortVolResults.M$Outcome            [SubCortVolResults.M$FDR <0.05],
    outcomesLab   =  SubCortVolResults.M$Label              [SubCortVolResults.M$FDR <0.05],
    ROI           =  SubCortVolResults.M$ElementDescription [SubCortVolResults.M$FDR <0.05],
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.m
  )

SubCortVol_SigResults_bySite.M      <- 
  SubCortVol_SigResults_bySite.M[complete.cases(SubCortVol_SigResults_bySite.M),]#filtering GLMEs that didnt fit 

SubCortVol_SigResults_bySite.M$Site <- 
  abcd$abcd_site[match(SubCortVol_SigResults_bySite.M$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Cortical Thickness by site Girls
CortThickResults_bySite.F <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortThickResults.F$Outcome            [CortThickResults.F$FDR <0.05],
    outcomesLab   =  CortThickResults.F$Label              [CortThickResults.F$FDR <0.05],
    ROI           =  CortThickResults.F$ElementDescription [CortThickResults.F$FDR <0.05],
    covars = c(covars.x, "smri_thick_cdk_mean_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.f
  )

CortThickResults_bySite.F      <- 
  CortThickResults_bySite.F[complete.cases(CortThickResults_bySite.F),]#filtering GLMEs that didnt fit 

CortThickResults_bySite.F$Site <- 
  abcd$abcd_site[match(CortThickResults_bySite.F$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Cortical Thickness by site Boys
CortThickResults_bySite.M <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortThickResults.M$Outcome            [CortThickResults.M$FDR <0.05],
    outcomesLab   =  CortThickResults.M$Label              [CortThickResults.M$FDR <0.05],
    ROI           =  CortThickResults.M$ElementDescription [CortThickResults.M$FDR <0.05],
    covars = c(covars.x, "smri_thick_cdk_mean_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.m
  )

CortThickResults_bySite.M      <- 
  CortThickResults_bySite.M[complete.cases(CortThickResults_bySite.M),]#filtering GLMEs that didnt fit 

CortThickResults_bySite.M$Site <- 
  abcd$abcd_site[match(CortThickResults_bySite.M$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Cortical Area by site Girls
CortAreaResults_bySite.F <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortAreaResults.F$Outcome            [CortAreaResults.F$FDR <0.05],
    outcomesLab   =  CortAreaResults.F$Label              [CortAreaResults.F$FDR <0.05],
    ROI           =  CortAreaResults.F$ElementDescription [CortAreaResults.F$FDR <0.05],
    covars = c(covars.x, "smri_area_cdk_total_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.f
  )

CortAreaResults_bySite.F      <- 
  CortAreaResults_bySite.F[complete.cases(CortAreaResults_bySite.F),]#filtering GLMEs that didnt fit 

CortAreaResults_bySite.F$Site <- 
  abcd$abcd_site[match(CortAreaResults_bySite.F$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Cortical Area by site Boys 
CortAreaResults_bySite.M <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  CortAreaResults.M$Outcome            [CortAreaResults.M$FDR <0.05],
    outcomesLab   =  CortAreaResults.M$Label              [CortAreaResults.M$FDR <0.05],
    ROI           =  CortAreaResults.M$ElementDescription [CortAreaResults.M$FDR <0.05],
    covars = c(covars.x, "smri_area_cdk_total_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.m
  )

CortAreaResults_bySite.M      <- 
  CortAreaResults_bySite.M[complete.cases(CortAreaResults_bySite.M),]#filtering GLMEs that didnt fit 

#Sulcal Depth by site Girls
SulcalDepthResults_bySite.F <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  SulcalDepthResults.F$Outcome               [SulcalDepthResults.F$FDR <0.05],
    outcomesLab   =  SulcalDepthResults.F$Label                 [SulcalDepthResults.F$FDR <0.05],
    ROI           =  SulcalDepthResults.F$ElementDescription    [SulcalDepthResults.F$FDR <0.05],
    covars = c(covars.x, "smri_sulc_cdk_mean_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.f
  )

SulcalDepthResults_bySite.F      <- 
  SulcalDepthResults_bySite.F[complete.cases(SulcalDepthResults_bySite.F),]#filtering GLMEs that didnt fit 

SulcalDepthResults_bySite.F$Site <- 
  abcd$abcd_site[match(SulcalDepthResults_bySite.F$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

#Sulcal Depth by site Boys
SulcalDepthResults_bySite.M <-
  LME_bySite_loop(
    predictors    = "A_Prenatal_Burden_Score_z",
    outcomes      =  SulcalDepthResults.M$Outcome               [SulcalDepthResults.M$FDR <0.05],
    outcomesLab   =  SulcalDepthResults.M$Label                 [SulcalDepthResults.M$FDR <0.05],
    ROI           =  SulcalDepthResults.M$ElementDescription    [SulcalDepthResults.M$FDR <0.05],
    covars = c(covars.x, "smri_sulc_cdk_mean_z"),
    Site =  "mri_info_deviceserialnumber",
    data = abcd.m
  )

SulcalDepthResults_bySite.M      <- 
  SulcalDepthResults_bySite.M[complete.cases(SulcalDepthResults_bySite.M),]#filtering GLMEs that didnt fit 

SulcalDepthResults_bySite.M$Site <- 
  abcd$abcd_site[match(SulcalDepthResults_bySite.M$Site, abcd$mri_info_deviceserialnumber)]# Deidentifying MRI

```

## Performing meta-analysis
```{r meta analysis, message=F, warning=FALSE, cache=TRUE}
#Selecting Significant ROIs
CortVol.Sig.F     <-  CortVolResults.F$Outcome   [CortVolResults.F$FDR <0.05]
SubCortVol.Sig.F  <-  SubCortVolResults.F$Outcome[SubCortVolResults.F$FDR <0.05]
CortThick.Sig.F   <-  CortThickResults.F$Outcome [CortThickResults.F$FDR <0.05]
CortArea.Sig.F    <-  CortAreaResults.F$Outcome  [CortAreaResults.F$FDR <0.05]
SulcalDepth.Sig.F <-  SulcalDepthResults.F$Outcome[SulcalDepthResults.F$FDR <0.05]

CortVol.Sig.M     <-  CortVolResults.M$Outcome   [CortVolResults.M$FDR <0.05]
SubCortVol.Sig.M  <-  SubCortVolResults.M$Outcome[SubCortVolResults.M$FDR <0.05]
CortThick.Sig.M   <-  CortThickResults.M$Outcome [CortThickResults.M$FDR <0.05]
CortArea.Sig.M    <-  CortAreaResults.M$Outcome  [CortAreaResults.M$FDR <0.05]
SulcalDepth.Sig.M <-  SulcalDepthResults.M$Outcome[SulcalDepthResults.M$FDR <0.05]

# Cortical Vol meta-analysis - Girls #
CortVol.Meta.F           <- as.data.frame(matrix(nrow = length(CortVol.Sig.F),ncol = 5))
colnames(CortVol.Meta.F) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortVol.Sig.F){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = CortVol_SigResults_bySite.F[CortVol_SigResults_bySite.F$Outcome==i,]$Beta_Predictor,
    sei = CortVol_SigResults_bySite.F[CortVol_SigResults_bySite.F$Outcome==i,]$Std.error,
    method = "FE")
  
  CortVol.Meta.F[iID,"ROI"]        <- i
  CortVol.Meta.F[iID,"PooledBeta"] <- meta.res$beta
  CortVol.Meta.F[iID,"p.value"]    <- meta.res$pval
  CortVol.Meta.F[iID,"ci.lb"]      <- meta.res$ci.lb
  CortVol.Meta.F[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Cortical Vol meta-analysis - Boys #
CortVol.Meta.M           <- as.data.frame(matrix(nrow = length(CortVol.Sig.M),ncol = 5))
colnames(CortVol.Meta.M) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortVol.Sig.M){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = CortVol_SigResults_bySite.M[CortVol_SigResults_bySite.M$Outcome==i,]$Beta_Predictor,
    sei = CortVol_SigResults_bySite.M[CortVol_SigResults_bySite.M$Outcome==i,]$Std.error,
    method = "FE")
  
  CortVol.Meta.M[iID,"ROI"]        <- i
  CortVol.Meta.M[iID,"PooledBeta"] <- meta.res$beta
  CortVol.Meta.M[iID,"p.value"]    <- meta.res$pval
  CortVol.Meta.M[iID,"ci.lb"]      <- meta.res$ci.lb
  CortVol.Meta.M[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Subcortical Volume - Girls #

SubCortVol.Meta.F           <- as.data.frame(matrix(nrow = length(SubCortVol.Sig.F),ncol = 5))
colnames(SubCortVol.Meta.F) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in SubCortVol.Sig.F){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = SubCortVol_SigResults_bySite.F[SubCortVol_SigResults_bySite.F$Outcome==i,]$Beta_Predictor,
    sei = SubCortVol_SigResults_bySite.F[SubCortVol_SigResults_bySite.F$Outcome==i,]$Std.error,
    method = "FE")
  
  SubCortVol.Meta.F[iID,"ROI"]        <- i
  SubCortVol.Meta.F[iID,"PooledBeta"] <- meta.res$beta
  SubCortVol.Meta.F[iID,"p.value"]    <- meta.res$pval
  SubCortVol.Meta.F[iID,"ci.lb"]      <- meta.res$ci.lb
  SubCortVol.Meta.F[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Subcortical Volume - Boys #

SubCortVol.Meta.M           <- as.data.frame(matrix(nrow = length(SubCortVol.Sig.M),ncol = 5))
colnames(SubCortVol.Meta.M) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in SubCortVol.Sig.M){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = SubCortVol_SigResults_bySite.M[SubCortVol_SigResults_bySite.M$Outcome==i,]$Beta_Predictor,
    sei = SubCortVol_SigResults_bySite.M[SubCortVol_SigResults_bySite.M$Outcome==i,]$Std.error,
    method = "FE")
  
  SubCortVol.Meta.M[iID,"ROI"]        <- i
  SubCortVol.Meta.M[iID,"PooledBeta"] <- meta.res$beta
  SubCortVol.Meta.M[iID,"p.value"]    <- meta.res$pval
  SubCortVol.Meta.M[iID,"ci.lb"]      <- meta.res$ci.lb
  SubCortVol.Meta.M[iID,"ci.ub"]      <- meta.res$ci.ub
}


# Cortical Thickness meta-analysis - Girls #
CortThick.Meta.F           <- as.data.frame(matrix(nrow = length(CortThick.Sig.F),ncol = 5))
colnames(CortThick.Meta.F) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortThick.Sig.F){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = CortThickResults_bySite.F[CortThickResults_bySite.F$Outcome==i,]$Beta_Predictor,
    sei = CortThickResults_bySite.F[CortThickResults_bySite.F$Outcome==i,]$Std.error,
    method = "FE")
  
  CortThick.Meta.F[iID,"ROI"]        <- i
  CortThick.Meta.F[iID,"PooledBeta"] <- meta.res$beta
  CortThick.Meta.F[iID,"p.value"]    <- meta.res$pval
  CortThick.Meta.F[iID,"ci.lb"]      <- meta.res$ci.lb
  CortThick.Meta.F[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Cortical Thickness meta-analysis - Boys #
CortThick.Meta.M           <- as.data.frame(matrix(nrow = length(CortThick.Sig.M),ncol = 5))
colnames(CortThick.Meta.M) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortThick.Sig.M){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi = CortThickResults_bySite.M[CortThickResults_bySite.M$Outcome==i,]$Beta_Predictor,
    sei = CortThickResults_bySite.M[CortThickResults_bySite.M$Outcome==i,]$Std.error,
    method = "FE")
  
  CortThick.Meta.M[iID,"ROI"]        <- i
  CortThick.Meta.M[iID,"PooledBeta"] <- meta.res$beta
  CortThick.Meta.M[iID,"p.value"]    <- meta.res$pval
  CortThick.Meta.M[iID,"ci.lb"]      <- meta.res$ci.lb
  CortThick.Meta.M[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Cortical Area meta-analysis - Girls #
CortArea.Meta.F           <- as.data.frame(matrix(nrow = length(CortArea.Sig.F),ncol = 5))
colnames(CortArea.Meta.F) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortArea.Sig.F){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi =  CortAreaResults_bySite.F[CortAreaResults_bySite.F$Outcome==i,]$Beta_Predictor,
    sei = CortAreaResults_bySite.F[CortAreaResults_bySite.F$Outcome==i,]$Std.error,
    method = "FE")
  
  CortArea.Meta.F[iID,"ROI"]        <- i
  CortArea.Meta.F[iID,"PooledBeta"] <- meta.res$beta
  CortArea.Meta.F[iID,"p.value"]    <- meta.res$pval
  CortArea.Meta.F[iID,"ci.lb"]      <- meta.res$ci.lb
  CortArea.Meta.F[iID,"ci.ub"]      <- meta.res$ci.ub
}

# Cortical Area meta-analysis - Boys
CortArea.Meta.M           <- as.data.frame(matrix(nrow = length(CortArea.Sig.M),ncol = 5))
colnames(CortArea.Meta.M) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in CortArea.Sig.M){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi =  CortAreaResults_bySite.M[CortAreaResults_bySite.M$Outcome==i,]$Beta_Predictor,
    sei = CortAreaResults_bySite.M[CortAreaResults_bySite.M$Outcome==i,]$Std.error,
    method = "FE")
  
  CortArea.Meta.M[iID,"ROI"]        <- i
  CortArea.Meta.M[iID,"PooledBeta"] <- meta.res$beta
  CortArea.Meta.M[iID,"p.value"]    <- meta.res$pval
  CortArea.Meta.M[iID,"ci.lb"]      <- meta.res$ci.lb
  CortArea.Meta.M[iID,"ci.ub"]      <- meta.res$ci.ub
}

# SulcalDepth meta-analysis - Girls #
SulcalDepth.Meta.F           <- as.data.frame(matrix(nrow = length(SulcalDepth.Sig.F),ncol = 5))
colnames(SulcalDepth.Meta.F) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in SulcalDepth.Sig.F){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi  = SulcalDepthResults_bySite.F[SulcalDepthResults_bySite.F$Outcome==i,]$Beta_Predictor,
    sei = SulcalDepthResults_bySite.F[SulcalDepthResults_bySite.F$Outcome==i,]$Std.error,
    method = "FE")
  
  SulcalDepth.Meta.F[iID,"ROI"]        <- i
  SulcalDepth.Meta.F[iID,"PooledBeta"] <- meta.res$beta
  SulcalDepth.Meta.F[iID,"p.value"]    <- meta.res$pval
  SulcalDepth.Meta.F[iID,"ci.lb"]      <- meta.res$ci.lb
  SulcalDepth.Meta.F[iID,"ci.ub"]      <- meta.res$ci.ub
}

#SulcalDepth meta-analysis meta-analysis - Boys #
SulcalDepth.Meta.M           <- as.data.frame(matrix(nrow = length(SulcalDepth.Sig.M),ncol = 5))
colnames(SulcalDepth.Meta.M) <- c("ROI",'PooledBeta','p.value','ci.lb','ci.ub')

iID <- 0
for( i in SulcalDepth.Sig.M){
  
  iID <- iID+1
  meta.res <- metafor::rma(
    yi  = SulcalDepthResults_bySite.M[SulcalDepthResults_bySite.M$Outcome==i,]$Beta_Predictor,
    sei = SulcalDepthResults_bySite.M[SulcalDepthResults_bySite.M$Outcome==i,]$Std.error,
    method = "FE")
  
  SulcalDepth.Meta.M[iID,"ROI"]        <- i
  SulcalDepth.Meta.M[iID,"PooledBeta"] <- meta.res$beta
  SulcalDepth.Meta.M[iID,"p.value"]    <- meta.res$pval
  SulcalDepth.Meta.M[iID,"ci.lb"]      <- meta.res$ci.lb
  SulcalDepth.Meta.M[iID,"ci.ub"]      <- meta.res$ci.ub
}

```

## Create Figure 4 - Meta-analysis
```{r, message=FALSE, warning=FALSE, fig.height=6, fig.width=12 }
# Girls
CortVol.Meta.F$Label                 <-CortVolResults.F[CortVolResults.F$Outcome%in%CortVol.Sig.F,"Label"]
CortVol.Meta.F$ElementDescription    <-CortVolResults.F[CortVolResults.F$Outcome%in%CortVol.Sig.F,"ElementDescription"]
CortVol.Meta.F$Std.Beta              <- CortVolResults.F[CortVolResults.F$Outcome%in%CortVol.Sig.F,"Beta_Predictor"]
CortVol.Meta.F$Std_p.value           <- CortVolResults.F[CortVolResults.F$Outcome%in%CortVol.Sig.F,"FDR"]

SubCortVol.Meta.F$Label              <- SubCortVolResults.F[SubCortVolResults.F$Outcome%in%SubCortVol.Sig.F ,"Label"]
SubCortVol.Meta.F$ElementDescription <- SubCortVolResults.F[SubCortVolResults.F$Outcome%in%SubCortVol.Sig.F,"ElementDescription"]
SubCortVol.Meta.F$Std.Beta           <- SubCortVolResults.F[SubCortVolResults.F$Outcome%in%SubCortVol.Sig.F,"Beta_Predictor"]
SubCortVol.Meta.F$Std_p.value        <- SubCortVolResults.F[SubCortVolResults.F$Outcome%in%SubCortVol.Sig.F,"FDR"]

CortThick.Meta.F$Label               <- CortThickResults.F[CortThickResults.F$Outcome%in%CortThick.Sig.F,"Label"]
CortThick.Meta.F$ElementDescription  <-CortThickResults.F[CortThickResults.F$Outcome%in%CortThick.Sig.F,"ElementDescription"]
CortThick.Meta.F$Std.Beta            <- CortThickResults.F[CortThickResults.F$Outcome%in%CortThick.Sig.F,"Beta_Predictor"]
CortThick.Meta.F$Std_p.value         <- CortThickResults.F[CortThickResults.F$Outcome%in%CortThick.Sig.F,"FDR"]

CortArea.Meta.F$Label                <- CortAreaResults.F[CortAreaResults.F$Outcome%in%CortArea.Sig.F,"Label"]
CortArea.Meta.F$ElementDescription   <- CortAreaResults.F[CortAreaResults.F$Outcome%in%CortArea.Sig.F,"ElementDescription"]
CortArea.Meta.F$Std.Beta             <- CortAreaResults.F[CortAreaResults.F$Outcome%in%CortArea.Sig.F,"Beta_Predictor"]
CortArea.Meta.F$Std_p.value          <- CortAreaResults.F[CortAreaResults.F$Outcome%in%CortArea.Sig.F,"FDR"]

SulcalDepth.Meta.F$Label              <- SulcalDepthResults.F[SulcalDepthResults.F$Outcome%in%SulcalDepth.Sig.F,"Label"]
SulcalDepth.Meta.F$ElementDescription <- SulcalDepthResults.F[SulcalDepthResults.F$Outcome%in%SulcalDepth.Sig.F,"ElementDescription"]
SulcalDepth.Meta.F$Std.Beta           <- SulcalDepthResults.F[SulcalDepthResults.F$Outcome%in%SulcalDepth.Sig.F,"Beta_Predictor"]
SulcalDepth.Meta.F$Std_p.value        <- SulcalDepthResults.F[SulcalDepthResults.F$Outcome%in%SulcalDepth.Sig.F,"FDR"]

Meta_Betas.F <- rbind(CortVol.Meta.F, SubCortVol.Meta.F, CortThick.Meta.F, CortArea.Meta.F, SulcalDepth.Meta.F)

# Boys
CortVol.Meta.M$Label                 <- CortVolResults.M[CortVolResults.M$Outcome%in%CortVol.Sig.M,"Label"]
CortVol.Meta.M$ElementDescription    <- CortVolResults.M[CortVolResults.M$Outcome%in%CortVol.Sig.M,"ElementDescription"]
CortVol.Meta.M$Std.Beta              <- CortVolResults.M[CortVolResults.M$Outcome%in%CortVol.Sig.M,"Beta_Predictor"]
CortVol.Meta.M$Std_p.value           <- CortVolResults.M[CortVolResults.M$Outcome%in%CortVol.Sig.M,"FDR"]

SubCortVol.Meta.M$Label              <- SubCortVolResults.M[SubCortVolResults.M$Outcome%in%SubCortVol.Sig.M ,"Label"]
SubCortVol.Meta.M$ElementDescription <- SubCortVolResults.M[SubCortVolResults.M$Outcome%in%SubCortVol.Sig.M,"ElementDescription"]
SubCortVol.Meta.M$Std.Beta           <- SubCortVolResults.M[SubCortVolResults.M$Outcome%in%SubCortVol.Sig.M,"Beta_Predictor"]
SubCortVol.Meta.M$Std_p.value        <- SubCortVolResults.M[SubCortVolResults.M$Outcome%in%SubCortVol.Sig.M,"FDR"]

CortThick.Meta.M$Label               <- CortThickResults.M[CortThickResults.M$Outcome%in%CortThick.Sig.M,"Label"]
CortThick.Meta.M$ElementDescription  <- CortThickResults.M[CortThickResults.M$Outcome%in%CortThick.Sig.M,"ElementDescription"]
CortThick.Meta.M$Std.Beta            <- CortThickResults.M[CortThickResults.M$Outcome%in%CortThick.Sig.M,"Beta_Predictor"]
CortThick.Meta.M$Std_p.value         <- CortThickResults.M[CortThickResults.M$Outcome%in%CortThick.Sig.M,"FDR"]

CortArea.Meta.M$Label                <- CortAreaResults.M[CortAreaResults.M$Outcome%in%CortArea.Sig.M,"Label"]
CortArea.Meta.M$ElementDescription   <- CortAreaResults.M[CortAreaResults.M$Outcome%in%CortArea.Sig.M,"ElementDescription"]
CortArea.Meta.M$Std.Beta             <- CortAreaResults.M[CortAreaResults.M$Outcome%in%CortArea.Sig.M,"Beta_Predictor"]
CortArea.Meta.M$Std_p.value          <- CortAreaResults.M[CortAreaResults.M$Outcome%in%CortArea.Sig.M,"FDR"]

SulcalDepth.Meta.M$Label              <- SulcalDepthResults.M[SulcalDepthResults.M$Outcome%in%SulcalDepth.Sig.M,"Label"]
SulcalDepth.Meta.M$ElementDescription <- SulcalDepthResults.M[SulcalDepthResults.M$Outcome%in%SulcalDepth.Sig.M,"ElementDescription"]
SulcalDepth.Meta.M$Std.Beta           <- SulcalDepthResults.M[SulcalDepthResults.M$Outcome%in%SulcalDepth.Sig.M,"Beta_Predictor"]
SulcalDepth.Meta.M$Std_p.value        <- SulcalDepthResults.M[SulcalDepthResults.M$Outcome%in%SulcalDepth.Sig.M,"FDR"]

Meta_Betas.M <- rbind(CortVol.Meta.M, SubCortVol.Meta.M,CortThick.Meta.M, SulcalDepth.Meta.M)#CortArea.Meta.M not sig

plotMeta.M <- ggplot(data = Meta_Betas.M,aes(y = reorder(ElementDescription, PooledBeta),x =  PooledBeta))+
    geom_point(shape=23, fill="black",size=3)+
    geom_errorbar(aes(xmin=ci.lb, xmax=ci.ub), colour="black", width=.1)+
    labs(x="Pooled Beta", y="Brain Data Modalities and Regions", title = 'Males' )+
    geom_vline(xintercept = 0, size=1.25)+
    coord_cartesian(xlim = c(-.14,.14))+
    theme_bw()+
    theme(axis.text.y = element_text(size = 9),axis.text.x = element_text(size = 7.5),
          plot.title = element_text(size = 11,hjust = 0.5),axis.title.x =  element_text(size = 8))

plotMeta.F <- ggplot(data = Meta_Betas.F,aes(y = reorder(ElementDescription, PooledBeta),x =  PooledBeta))+
    geom_point(shape=23, fill="black",size=3)+
    geom_errorbar(aes(xmin=ci.lb, xmax=ci.ub), colour="black", width=.1)+
    labs(x="Pooled Beta", y="Brain Data Modalities and Regions", title = 'Females')+
    geom_vline(xintercept = 0, size=1.25)+
    coord_cartesian(xlim = c(-.14,.14))+
    theme_bw()+
    theme(axis.text.y = element_text(size = 9),axis.text.x = element_text(size = 7.5),
          plot.title = element_text(size = 11,hjust = 0.5),axis.title.x =  element_text(size = 8))

grid.arrange(plotMeta.F, plotMeta.M, ncol=2)

tiff(filename = "Figure4", height = 6,width = 12, units = 'in', res = 1200, compression = "zip")
grid.arrange(plotMeta.F, plotMeta.M, ncol=2)
dev.off()

```


# eTable 1 - Comparison of study variables between full sample and complete cases sample
## Prepare results for eTable 1

```{r}

abcd_full$brainvol.pgs_z <- scale(abcd_full$brainvol.pgs) %>% as.numeric()
num.descriptives.full <- mean.describe2(x = abcd_full[,
                        c("A_Prenatal_Burden_Score",
                          'interview_age'      ,# Child‚Äôs age
                          'demo_comb_income_v2',# Combined income
                          'demo_caregiver_ed'  ,# Averaged caregivers education
                          'fes_fam_env_score'  ,# Family Conflict Subscale - Family Environment Scale
                          'devhx_3_p'          ,# Maternal age at birth
                          'asr_scr_totprob_t'  ,# Parent Total Problems - ASEBA
                          'reshist_addr1_adi_wsum',
                          'Average_Screentime',
                          "brainvol.pgs_z")],
               group = abcd_full$sex
                )

rownames(num.descriptives.full) <- c("Prenatal Adverse Exposure Score, Mean (SD)",
                                "Age in months, Mean (SD)",
                                "Family combined income, Mean (SD)",
                                "Averaged carigiver's education, Mean (SD)",
                                "Family conflict subscale, Mean (SD)",
                                "Maternal age at birth, Mean (SD)",
                                "Parent Total Problems - ASEBA, Mean (SD)",
                                "Neighborhood safety, Mean (SD)",
                                "Average screen time use, Mean (SD)",
                                "Brain volume polygenic score, Mean (SD)"
                                )

colnames(num.descriptives.full) <- c("Females", 'Males')

# Running t. tests
t.values  <- c()
p.values <- c()
iID <- 0
for( i in c("A_Prenatal_Burden_Score",
            'interview_age'      ,# child‚Äôs age
            'demo_comb_income_v2',# Combined income
            'demo_caregiver_ed'  ,# Averaged caregivers education
            'fes_fam_env_score'  ,# Family Conflict Subscale - Family Environment Scale
            'devhx_3_p'          ,# Maternal age at birth
            'asr_scr_totprob_t'  ,# Parent Total Problems - ASEBA
            'neighborhood_safety',
            'Average_Screentime',
            'brainvol.pgs_z')){
  iID <- iID+1
  p.values[iID] <- t.test(abcd_full[,i] ~ abcd_full$sex, var.equal=T)$p.value
  t.values[iID] <- t.test(abcd_full[,i] ~ abcd_full$sex, var.equal=T)$statistic
}

num.descriptives.full$stat.value <- t.values
num.descriptives.full$p.value <- p.values

prop.descriptives.full.race              <- by( abcd_full[,c('demo_race_ethnicity')],abcd_full$sex,propfunc )
prop.descriptives.full.race              <- cbind(prop.descriptives.full.race[[1]],prop.descriptives.full.race[[2]]) #List to matrix
prop.descriptives.full.race              <- prop.descriptives.full.race[,c(2:3,5:6)]# Selecting variables, values and descriptives
prop.descriptives.full.race$Females      <- paste0(prop.descriptives.full.race$N, " (", prop.descriptives.full.race$`%`,")")#
prop.descriptives.full.race$Males        <- paste0(prop.descriptives.full.race$N.1, " (", prop.descriptives.full.race$`%.1`,")")
prop.descriptives.full.race              <- rbind(NA,prop.descriptives.full.race)
rownames(prop.descriptives.full.race)[1] <- "Ethnicity, N (%)"
prop.descriptives.full.race              <- prop.descriptives.full.race[,c("Females", 'Males')]
prop.descriptives.full.race$stat.value   <- NA
prop.descriptives.full.race$p.value      <- NA
prop.descriptives.full.race$stat.value[1]<- chisq.test(abcd_full$demo_race_ethnicity, abcd_full$sex)$statistic
prop.descriptives.full.race$p.value[1]   <- chisq.test(abcd_full$demo_race_ethnicity, abcd_full$sex)$p.value 


## Preparing Labels
Prenatal.vars <- Hmisc::Cs(devhx_6_p,               # Unplanned pregnancy
                           devhx_13_3_p,            # C-section 
                           A_Burden_Gestational_Age,
                           A_index_birth_weight,
                           devhx_8_tobacco,
                           devhx_8_alcohol,
                           devhx_8_marijuana,
                           A_Burden_Pregnan_Complications,
                           A_Burden_Birth_Complications)

prop.descriptives.full.PrenatalAdversity              <- by( abcd_full[,Prenatal.vars],abcd_full$sex,propfunc )
prop.descriptives.full.PrenatalAdversity              <- cbind(prop.descriptives.full.PrenatalAdversity[[1]],prop.descriptives.full.PrenatalAdversity[[2]]) #List to matrix
prop.descriptives.full.PrenatalAdversity              <- prop.descriptives.full.PrenatalAdversity[seq(2,18, by=2),c(3:4,7:8)]# Selecting variables, values and descriptives
prop.descriptives.full.PrenatalAdversity$Females      <- paste0(prop.descriptives.full.PrenatalAdversity$N, " (", prop.descriptives.full.PrenatalAdversity$`%`,")")#
prop.descriptives.full.PrenatalAdversity$Males        <- paste0(prop.descriptives.full.PrenatalAdversity$N.1, " (", prop.descriptives.full.PrenatalAdversity$`%.1`,")")
prop.descriptives.full.PrenatalAdversity              <- rbind(NA,prop.descriptives.full.PrenatalAdversity)
rownames(prop.descriptives.full.PrenatalAdversity)[1] <- "Adverse Prenatal Exposures, N (%)"
prop.descriptives.full.PrenatalAdversity              <- prop.descriptives.full.PrenatalAdversity[,c("Females", 'Males')]
prop.descriptives.full.PrenatalAdversity$stat.value   <- NA
prop.descriptives.full.PrenatalAdversity$p.value      <- NA

iID <- 1
for( i in Prenatal.vars){
    iID <- iID+1
    prop.descriptives.full.PrenatalAdversity$stat.value[iID]<- chisq.test(abcd_full[,i], abcd_full$sex)$statistic
    prop.descriptives.full.PrenatalAdversity$p.value[iID]   <- chisq.test(abcd_full[,i], abcd_full$sex)$p.value     
    
}

Prenatal.labels <- c("Unplanned pregnancy",
                    "C-Section",
                    "Gestational Age",
                    "Birth weight",
                    "Tobacco use",
                    "Alcohol use",
                    "Marijuana use",
                    "Pregnancy complications",
                    "Birth complications")

rownames(prop.descriptives.full.PrenatalAdversity) <- c("Adverse Prenatal Exposures, N (%)", Prenatal.labels )

# Merging Descriptives
participants.descriptives.full      <- rbind(num.descriptives.full,prop.descriptives.full.race,prop.descriptives.full.PrenatalAdversity)

#Saving eTable1
write.xlsx(x = cbind(participants.descriptives,participants.descriptives.full),
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable1',
           row.names = T,
           showNA = F, 
           append = T
           )

```

## Create eTable 1
```{r}
kable(cbind(participants.descriptives,participants.descriptives.full),
      digits = 2 ,
      caption = "eTable 1 - Comparison of study variables between full sample and complete cases sample")  %>% 
  add_header_above(c(" "= 1, "Study sample" = 4, "ABCD full sample"=4)) %>% 
  kableExtra::kable_styling(position = "center")
```

# eTable2 - List of Associations between Prenatal Adverse Exposures and Brain Regions

## Prepare results for eTable2

```{r}

#Saving eTable2
write.xlsx(x = cbind(Main_results.F[,-c(1,3)], Main_results.M[,6:11]),#Omit Main effect ID and Aliases
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable2',
           row.names = F,
           showNA = F, 
           append = T
           )

```

## Create eTable2
```{r}
kable( cbind(Main_results.F[,-c(1,3)], Main_results.M[,6:11]),#Omit Main effect ID and Aliases
      caption = 'eTable 2 - List of Associations between Prenatal Adverse Exposures and Brain Regions',
      digits = 3
      ) %>% 
  add_header_above(c(" "= 3, "Females" = 6, "Males"=6)) %>% 
  kable_styling(position = "center")


```

# eTable3 - Associations Between Brain Components with Behavior Problems and Cognition
## Saving eTable3
```{r}

write.xlsx(x = cbind(CBCL_results.F[,-c(1,3:4)],CBCL_results.M[,-c(1:5)]),
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable3',
           row.names = T,
           showNA = F, 
           append = T
           )



```

# eTable4 - Associations Between Prenatal Adversity with Children's Behavior and Cognition
## Saving eTable4
```{r}

                                                        
write.xlsx(x = cbind(Prenatal_results.F[,-c(1,3:4)],Prenatal_results.M[,-c(1:5)]),
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable4',
           row.names = T,
           showNA = F, 
           append = T
           )

                                                        
```

# eTable5 - Associations between Prenatal Adverse Exposures and Significant Brain Regions in Females by Site 
## Prepare results for eTable5

```{r}

Main_results_bySite.F                               <- rbind(CortVol_SigResults_bySite.F,
                                                             CortThickResults_bySite.F,
                                                             CortAreaResults_bySite.F,
                                                             SulcalDepthResults_bySite.F)
#Saving eTable5
write.xlsx(x = Main_results_bySite.F[complete.cases(Main_results_bySite.F),-c(1,4)],#Omit Main effect ID and Aliases and sites in which glme did not fit
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable5',
           row.names = F,
           showNA = F, 
           append = T
           )

```

## Create eTable5
```{r}
kable(Main_results_bySite.F[complete.cases(Main_results_bySite.F),-c(1,4)],#Omit Main effect ID and Aliases and sites in which glme did not fit
      caption = 'eTable5 - Associations between Prenatal Adverse Exposures and Significant Brain Regions in Females by Site',
      digits = 3) %>% 
  kable_styling(position = "center")

```

# eTable6 - Associations between Prenatal Adverse Exposures and Significant Brain Regions in Males by Site 
## Prepare results for eTable6

```{r}

Main_results_bySite.M                           <- rbind(CortVol_SigResults_bySite.M,
                                                         SubCortVol_SigResults_bySite.M,
                                                         CortThickResults_bySite.M,
                                                         SulcalDepthResults_bySite.M)
#Saving eTable6
write.xlsx(x = Main_results_bySite.M[complete.cases(Main_results_bySite.M),-c(1,4)],#Omit Main effect ID and Aliases and sites in which glme did not fit
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable6',
           row.names = F,
           showNA = F, 
           append = T
           )


```

## Create eTable6
```{r}
kable(Main_results_bySite.M[complete.cases(Main_results_bySite.M),-c(1,4)],#Omit Main effect ID and Aliases and sites in which glme did not fit
      caption = 'eTable6 - Associations between Prenatal Adverse Exposures and Significant Brain Regions in Males by Site',
      digits = 3) %>% 
  kable_styling(position = "center")

```

# Specificity analysis

# eTable7 - Associations between Individual Prenatal Adverse Exposures and Brain Regions
## Running analysis for each individual adverse exposures

```{r Sensitivity analysis Females, cache=TRUE}

#### Females ####
# Cortical Volume by exposure - Females 
CortVol_sensiResults.F <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CVol_outcomes,
    covars = c(covars.x, "smri_vol_scs_intracranialv_z"),
    Scale = TRUE,
    data = abcd.f
  )

CortVol_sensiResults.F$ElementDescription <- 
  CortVolResults.F$ElementDescription[match(CortVol_sensiResults.F$Outcome, CortVolResults.F$Outcome)]

CortVol_sensiResults.F$FDR <- NA
for( i in Prenatal.labels){
  CortVol_sensiResults.F[which(CortVol_sensiResults.F$Predictor==i),]$FDR <-
  p.adjust(
    CortVol_sensiResults.F[CortVol_sensiResults.F$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Subcortical Volume by exposure - Females
SubCortVol_sensiResults.F <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  SubCortical_outcomes,
    covars = c(covars.x, 'smri_vol_scs_intracranialv_z'),
    Scale = TRUE,
    data = abcd.f
  )

SubCortVol_sensiResults.F$ElementDescription <- 
  SubCortVolResults.F$ElementDescription[match(SubCortVol_sensiResults.F$Outcome, SubCortVolResults.F$Outcome)]

SubCortVol_sensiResults.F$FDR <- NA
for( i in Prenatal.labels){
  SubCortVol_sensiResults.F[which(SubCortVol_sensiResults.F$Predictor==i),]$FDR <-
  p.adjust(
    SubCortVol_sensiResults.F[SubCortVol_sensiResults.F$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}
# Cortical Thickness by exposure - Females
CortThick_sensiResults.F <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CortThick_outcomes, 
    covars = c(covars.x, 'smri_thick_cdk_mean_z'),
    Scale = TRUE,
    data = abcd.f
  )

CortThick_sensiResults.F$ElementDescription <- 
  CortThickResults.F$ElementDescription[match(CortThick_sensiResults.F$Outcome, CortThickResults.F$Outcome)]

CortThick_sensiResults.F$FDR <- NA
for( i in Prenatal.labels){
  CortThick_sensiResults.F[which(CortThick_sensiResults.F$Predictor==i),]$FDR <-
  p.adjust(
    CortThick_sensiResults.F[CortThick_sensiResults.F$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Cortical Area by exposure - Females  
CortArea_sensiResults.F <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CortArea_outcomes,
    covars = c(covars.x, 'smri_area_cdk_total_z'),
    Scale = TRUE,
    data = abcd.f
  )

CortArea_sensiResults.F$ElementDescription <- 
  CortAreaResults.F$ElementDescription[match(CortArea_sensiResults.F$Outcome, CortAreaResults.F$Outcome)]

CortArea_sensiResults.F$FDR <- NA
for( i in Prenatal.labels){
  CortArea_sensiResults.F[which(CortArea_sensiResults.F$Predictor==i),]$FDR <-
  p.adjust(
    CortArea_sensiResults.F[CortArea_sensiResults.F$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Sulcal Depth by exposure - Females  
SulcalDepth_sensiResults.F <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  SulcalDepth_outcomes,
    covars = c(covars.x,'smri_sulc_cdk_mean_z'),
    Scale = TRUE,
    data = abcd.f
  )

SulcalDepth_sensiResults.F$ElementDescription <- 
  SulcalDepthResults.F$ElementDescription[match(SulcalDepth_sensiResults.F$Outcome, SulcalDepthResults.F$Outcome)]

SulcalDepth_sensiResults.F$FDR <- NA
for( i in Prenatal.labels){
  SulcalDepth_sensiResults.F[which(SulcalDepth_sensiResults.F$Predictor==i),]$FDR <-
  p.adjust(
    SulcalDepth_sensiResults.F[SulcalDepth_sensiResults.F$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}


#### Males ####
# Cortical Volume by exposure - Males
CortVol_sensiResults.M <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CVol_outcomes,
    covars = c(covars.x,'smri_vol_scs_intracranialv_z'),
    Scale = TRUE,
    data = abcd.m
  )

CortVol_sensiResults.M$ElementDescription <- 
  CortVolResults.M$ElementDescription[match(CortVol_sensiResults.M$Outcome, CortVolResults.M$Outcome)]

CortVol_sensiResults.M$FDR <- NA
for( i in Prenatal.labels){
  CortVol_sensiResults.M[which(CortVol_sensiResults.M$Predictor==i),]$FDR <-
  p.adjust(
    CortVol_sensiResults.M[CortVol_sensiResults.M$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Subcortical Volume by exposure - Males
SubCortVol_sensiResults.M <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  SubCortical_outcomes,
    covars = c(covars.x, 'smri_vol_scs_intracranialv_z'),
    Scale = TRUE,
    data = abcd.m
  )

SubCortVol_sensiResults.M$ElementDescription <- 
  SubCortVolResults.M$ElementDescription[match(SubCortVol_sensiResults.M$Outcome, SubCortVolResults.M$Outcome)]

SubCortVol_sensiResults.M$FDR <- NA
for( i in Prenatal.labels){
  SubCortVol_sensiResults.M[which(SubCortVol_sensiResults.M$Predictor==i),]$FDR <-
  p.adjust(
    SubCortVol_sensiResults.M[SubCortVol_sensiResults.M$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Cortical Thickness by exposure - Males
CortThick_sensiResults.M <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CortThick_outcomes,
    covars = c(covars.x, 'smri_thick_cdk_mean_z'),
    Scale = TRUE,
    data = abcd.m
  )

CortThick_sensiResults.M$ElementDescription <- 
  CortThickResults.M$ElementDescription[match(CortThick_sensiResults.M$Outcome, CortThickResults.M$Outcome)]

CortThick_sensiResults.M$FDR <- NA
for( i in Prenatal.labels){
  CortThick_sensiResults.M[which(CortThick_sensiResults.M$Predictor==i),]$FDR <-
  p.adjust(
    CortThick_sensiResults.M[CortThick_sensiResults.M$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

# Cortical Area by exposure - Males  
CortArea_sensiResults.M <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  CortArea_outcomes,
    covars = c(covars.x,'smri_area_cdk_total_z'),
    Scale = TRUE,
    data = abcd.m
  )

CortArea_sensiResults.M$ElementDescription <- 
  CortAreaResults.M$ElementDescription[match(CortArea_sensiResults.M$Outcome, CortAreaResults.M$Outcome)]

CortArea_sensiResults.M$FDR <- NA
for( i in Prenatal.labels){
  CortArea_sensiResults.M[which(CortArea_sensiResults.M$Predictor==i),]$FDR <-
  p.adjust(
    CortArea_sensiResults.M[CortArea_sensiResults.M$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}


# Sulcal Depth by exposure - Males  
SulcalDepth_sensiResults.M <-
  LME_loop(
    predictors = Prenatal.vars,
    predictorsLab = Prenatal.labels,
    outcomes =  SulcalDepth_outcomes,
    covars = c(covars.x,'smri_sulc_cdk_mean_z'),
    Scale = TRUE,
    data = abcd.m
  )

SulcalDepth_sensiResults.M$ElementDescription <- 
  SulcalDepthResults.M$ElementDescription[match(SulcalDepth_sensiResults.M$Outcome, SulcalDepthResults.M$Outcome)]

SulcalDepth_sensiResults.M$FDR <- NA
for( i in Prenatal.labels){
  SulcalDepth_sensiResults.M[which(SulcalDepth_sensiResults.M$Predictor==i),]$FDR <-
  p.adjust(
    SulcalDepth_sensiResults.M[SulcalDepth_sensiResults.M$Predictor==i,]$p_value_Predictor,
    "fdr"
    )
}

```


## Prepare results for eTable7

```{r}

SensitivityResults.F <- rbind(CortVol_sensiResults.F,
      SubCortVol_sensiResults.F,
      CortThick_sensiResults.F,
      CortArea_sensiResults.F,
      SulcalDepth_sensiResults.F)

SensitivityResults.M <- rbind(CortVol_sensiResults.M,
      SubCortVol_sensiResults.M,
      CortThick_sensiResults.M,
      CortArea_sensiResults.M,
      SulcalDepth_sensiResults.M)


```

## Saving eTable7
```{r}
# Saving eTable7
write.xlsx(x = cbind(SensitivityResults.F[,-c(1,3)],  #Omit Id and label column
            SensitivityResults.M[,6:11]),             #Omit repeated columns,
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable7',
           row.names = F,
           showNA = F, 
           append = T
           )

```


## Create eTable7
```{r}

kable(cbind(SensitivityResults.F[,-c(1,3)],  #Omit Id and label column
            SensitivityResults.M[,6:11]),    #Omit repeated columns
      digits = 3,
      caption = "eTable7 - Associations between Individual Prenatal Adverse Exposures and Brain Regions") %>% 
  add_header_above(c(" "=3, 'Females'=6, 'Males'=6)) %>% 
  kable_styling(position = "center")

```

# eTable8
## Define LME moderation function
```{r}
LME_moderation_loop <-
  function(predictors,
           outcomes,
           moderators,
           covars = NULL,
           data,
           predictorsLab = NULL,
           outcomesLab = NULL,
           ROI=NULL,
           print.model = NULL,
           dig = NULL,
           Scale=NULL,
           is.log10 = FALSE
  ){
    require(dplyr)
    
    interac_names   <- c("MainEffect_ID",
                         "Outcome",
                         "Label",
                         'ElementDescription',
                         "Predictor",
                         "Beta_Predictor",
                         'Std.error',
                         "p_value_Predictor",
                         'Beta_Sex',
                         "Std.error_Sex",
                         "p_value_Sex",
                         'Beta_interaction',
                         'Std.error_interaction',
                         'p_value_interaction',
                         "Sample_size",
                         "R2"
                         )
    
    
    #Labels
    
    if(!is.null(predictorsLab)){
        
        predictors <-  predictors
        predictorsLabels <- data.frame(vars = predictors,
                                       names= predictorsLab
                                       ,stringsAsFactors = F)
    }
    
    if(!is.null(outcomesLab)){
        
        outcomes <-  outcomes
        outcomesLabels <- data.frame(vars = outcomes,
                                     names= outcomesLab,
                                     ROI=ROI,
                                     stringsAsFactors = F)
    }
    #Results 
    main_results <- matrix(NA, nrow = length(predictors)*length(outcomes), ncol = 16) %>% as.data.frame()
    colnames(main_results) <- interac_names
    
    #Preparing data
    
    if (is.null(covars)) {
        lm.formula <- "outcome.y ~ predictor.x * moderator.x + "
    } else{
        lm.formula <-
            paste("outcome.y ~ predictor.x * moderator.x + ", paste(covars, collapse = "+"))
    }
    if(is.log10==TRUE){lm.formula <- gsub("outcome.y", "log10(outcome.y)", lm.formula)}
    if(isTRUE(Scale))   {lm.formula <- gsub("outcome.y", "scale(outcome.y)", lm.formula)}
    
    
    iID <- 0
    for(i in outcomes){
        for(j in predictors){
            
            iID <- iID+1
            data$outcome.y    <- data[,i]  
            data$predictor.x  <- data[,j]
            data$moderator.x  <- data[,moderators]
            
            if (is.null(covars)) {
              data1 <- data[ complete.cases( data[,c("outcome.y", "predictor.x")] ), ]#covars
            } else{
              data1 <- data[ complete.cases( data[,c("outcome.y", "predictor.x", covars)] ), ]#covars
            }
            
            main_results$Sample_size   [iID] <- nrow(data1)
            main_results$MainEffect_ID [iID] <- iID
            main_results$Outcome       [iID] <- i
            
            if(!is.null(outcomesLab)){
                main_results$Label              [iID] <- outcomesLabels[outcomesLabels$vars== i,"names"]
                main_results$ElementDescription [iID] <- outcomesLabels[outcomesLabels$vars== i,"ROI"]
            }
            
            if(!is.null(predictorsLab)){
                main_results$Predictor [iID] <- predictorsLabels[predictorsLabels$vars== j,"names"]
            }else{main_results$Predictor  [iID] <- j}
            
            
            if(nrow(data1)<15)next
            
            formula.random="~(1|mri_info_deviceserialnumber/rel_family_id)"
            
            model.loop = gamm4::gamm4(
              as.formula(lm.formula),
              random = as.formula(formula.random),
              weights = NULL,
              data = data1
            )

            main_results$Beta_Predictor        [iID] <- summary(model.loop$gam)$p.table[2,1] 
            main_results$Std.error             [iID] <- summary(model.loop$gam)$p.table[2,2] 
            main_results$p_value_Predictor     [iID] <- summary(model.loop$gam)$p.table[2,4]
            
            main_results$Beta_Sex              [iID] <- summary(model.loop$gam)$p.table[3,1] 
            main_results$Std.error_Sex         [iID] <- summary(model.loop$gam)$p.table[3,2] 
            main_results$p_value_Sex           [iID] <- summary(model.loop$gam)$p.table[3,4] 
            
            
            interact_row <- nrow(summary(model.loop$gam)$p.table)
            main_results$Beta_interaction      [iID] <- summary(model.loop$gam)$p.table[interact_row,1] 
            main_results$Std.error_interaction [iID] <- summary(model.loop$gam)$p.table[interact_row,2] 
            main_results$p_value_interaction   [iID] <- summary(model.loop$gam)$p.table[interact_row,4] 
            
            
            main_results$R2                  [iID] <- summary(model.loop$gam)$r.sq %>% round(3)
            
            if (!is.null(print.model)) {
                print(i)
                print(j)
                print(summary(model.loop$gam)$p.table)
        }
        
    }
    
    }
    
    main_results <<- main_results
  }

```

## Running analysis
```{r Moderation, cache=TRUE}
ModerationResults_CVol <-
  LME_moderation_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    moderators = 'sex',
    outcomes =  CVol_outcomes,
    outcomesLab = CvolMRI_outcomes_Labels,
    ROI = CvolROIs,
    Scale = T,
    covars = c(covars.x,'smri_vol_scs_intracranialv_z'),
    data = rbind(abcd.f,abcd.m)
  )

ModerationResults_SubCorticalVol <-
  LME_moderation_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    moderators = 'sex',
    outcomes =  SubCortical_outcomes,
    outcomesLab = SubCortical_outcomes_Labels,
    ROI = subROIs,
    Scale = T,
    covars = c(covars.x,'smri_vol_scs_intracranialv_z'),
    data = rbind(abcd.f,abcd.m)
  )


ModerationResults_CortThick <-
  LME_moderation_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    moderators = 'sex',
    outcomes =  CortThick_outcomes,
    outcomesLab = CortThick_outcomes_Labels,
    ROI = CortThickROIs,
    Scale = T,
    covars = c(covars.x,'smri_thick_cdk_mean_z'),
    data = rbind(abcd.f,abcd.m)
  )

ModerationResults_CortArea <-
  LME_moderation_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    moderators = 'sex',
    outcomes =  CortArea_outcomes,
    outcomesLab = CortArea_outcomes_Labels,
    ROI = CortAreaROIs,
    Scale = T,
    covars = c(covars.x,'smri_area_cdk_total_z'),
    data = rbind(abcd.f,abcd.m)
  )

ModerationResults_SulcalDepth <-
  LME_moderation_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    moderators = 'sex',
    outcomes =  SulcalDepth_outcomes,
    outcomesLab = SulcalDepth_outcomes_Labels,
    ROI = SulcalDepthROIs,
    Scale = T,
    covars = c(covars.x,'smri_sulc_cdk_mean_z'),
    data = rbind(abcd.f,abcd.m)
  )

ModerationResults <- rbind(ModerationResults_CVol, 
                           ModerationResults_SubCorticalVol,
                           ModerationResults_CortThick,
                           ModerationResults_CortArea,
                           ModerationResults_SulcalDepth
                           )

ModerationResults$Data_Modality <- c(rep("Cortical volume", 68),
                                     rep('Subcortical volume', 20),
                                     rep("Cortical thickness", 68),
                                     rep("Cortical area", 68),
                                     rep("Sulcal depth", 68)
                                     )

ModerationResults$FDR <- NA

for( i in c("Cortical volume",
            'Subcortical volume',
            "Cortical thickness",
            "Cortical area",
            "Sulcal depth")
     ){
  ModerationResults[which(ModerationResults$Data_Modality==i),]$FDR <-
  p.adjust(
    ModerationResults[ModerationResults$Data_Modality==i,]$p_value_interaction,
    "fdr"
    )
}

# Saving eTable8
write.xlsx(x = ModerationResults[,-c(1, 3)],
           file = "eSupp_SexSpecific associations of adverse prenatal burden exposure in childrens brain.xlsx",
           sheetName = 'eTable8',
           row.names = F,
           showNA = F, 
           append = T
           )

```

## Create eTable8

```{r}
kable(ModerationResults[,-c(1, 3)],#Omit iID and Aliases
      caption = "eTable8. Interactions between Prenatal Adverse Exposures and Sex on Brain Structure",
      digits = 3,
      row.names = F)%>%
  kableExtra::kable_styling(position = "center")
```


# Summary of sensitivity analysis

**Cortical Volume**
\
Out of `r length(CortVol.Sig.F)` significant Cortical Volume ROIs associated with Prenatal Adversity in females, `r length( CortVol_sensiResults.F[(CortVol_sensiResults.F$FDR<0.05) & (CortVol_sensiResults.F$Outcome %in% CortVol.Sig.F), "Outcome"])` were also significantly associated with individual prenatal exposures.\
Overlapping ROIs: `r CortVol_sensiResults.F[CortVol_sensiResults.F$FDR<0.05 & (CortVol_sensiResults.F$Outcome %in% CortVol.Sig.F), 'ElementDescription' ]`.\
Prenatal Vars: `r CortVol_sensiResults.F[CortVol_sensiResults.F$FDR<0.05 & (CortVol_sensiResults.F$Outcome %in% CortVol.Sig.F), "Predictor" ]`
\
Out of `r length(CortVol.Sig.M)` significant Cortical Volume ROIs associated with Prenatal Adversity in males, only `r length(CortVol_sensiResults.M[CortVol_sensiResults.M$FDR<0.05 & (CortVol_sensiResults.M$Outcome %in% CortVol.Sig.M),"Outcome"])`were also significant with individual risks.\
Overlapping ROIs: `r CortVol_sensiResults.M[CortVol_sensiResults.M$FDR<0.05 & (CortVol_sensiResults.M$Outcome %in% CortVol.Sig.M),'ElementDescription']`.\
Prenatal vars:`r CortVol_sensiResults.M[CortVol_sensiResults.M$FDR<0.05 & (CortVol_sensiResults.M$Outcome %in% CortVol.Sig.M),"Predictor"]`\
\
\
**Subcortical Volume**
\
Out of `r length(SubCortVol.Sig.M)` significant Subcortical Volume ROIs associated with Prenatal Adversity in males, `r length(SubCortVol_sensiResults.M[SubCortVol_sensiResults.M$FDR<0.05 & (SubCortVol_sensiResults.M$Outcome %in% SubCortVol.Sig.M), "Outcome"])`were also significant with individual exposure.\
Overlapping ROIs:`r SubCortVol_sensiResults.M[SubCortVol_sensiResults.M$FDR<0.05 & (SubCortVol_sensiResults.M$Outcome %in% SubCortVol.Sig.M), "ElementDescription"]`\
Prenatal vars:`r SubCortVol_sensiResults.M[SubCortVol_sensiResults.M$FDR<0.05 & (SubCortVol_sensiResults.M$Outcome %in% SubCortVol.Sig.M), "Predictor"]`\
\
\
**Cortical Thickness**
\
Out of `r length(CortThick.Sig.F)` significant Cortical Thickness regions associated with Prenatal Adversity in females,  `r length(CortThick_sensiResults.F[CortThick_sensiResults.F$FDR<0.05 & (CortThick_sensiResults.F$Outcome %in% CortThick.Sig.F), "Outcome"])`were also significant.\
Overlaping ROIs:`r CortThick_sensiResults.F[CortThick_sensiResults.F$FDR<0.05 & (CortThick_sensiResults.F$Outcome %in% CortThick.Sig.F), 'ElementDescription']`\
Prenatal vars:`r CortThick_sensiResults.F[CortThick_sensiResults.F$FDR<0.05 & (CortThick_sensiResults.F$Outcome %in% CortThick.Sig.F), "Predictor"]`\
\
Out of `r length(CortThick.Sig.M)` significant Cortical Thickness regions associated with Prenatal Adversity in males,  `r length(CortThick_sensiResults.M[CortThick_sensiResults.M$FDR<0.05 & (CortThick_sensiResults.M$Outcome %in% CortThick.Sig.M), "ElementDescription"])` were also significant.\
Overlaping ROIs:`r CortThick_sensiResults.M[CortThick_sensiResults.M$FDR<0.05 & (CortThick_sensiResults.M$Outcome %in% CortThick.Sig.M), "ElementDescription"]`\
Prenatal vars:`r CortThick_sensiResults.M[CortThick_sensiResults.M$FDR<0.05 & (CortThick_sensiResults.M$Outcome %in% CortThick.Sig.M), "Predictor"]`\
\
\
**Cortical Area**
\
Out of `r length(CortArea.Sig.F)` significant Cortical Area regions associated with Prenatal Adversity infemales, only `r length(CortArea_sensiResults.F[CortArea_sensiResults.F$FDR<0.05 & (CortArea_sensiResults.F$Outcome %in% CortArea.Sig.F), "Outcome"])`.\
Overlaping ROIs:`r CortArea_sensiResults.F[CortArea_sensiResults.F$FDR<0.05 & (CortArea_sensiResults.F$Outcome %in% CortArea.Sig.F), "ElementDescription"]`\
Prenatal vars:`r CortArea_sensiResults.F[CortArea_sensiResults.F$FDR<0.05 & (CortArea_sensiResults.F$Outcome %in% CortArea.Sig.F), "Predictor"]`\
\
Out of `r length(CortArea.Sig.M)` significant Cortical Area regions associated with Prenatal Adversity in males, only `r length(CortArea_sensiResults.M[CortArea_sensiResults.M$FDR<0.05 & (CortArea_sensiResults.M$Outcome %in% CortArea.Sig.M), "Outcome"])` were significant.\
Overlaping ROIs:`r CortArea_sensiResults.M[CortArea_sensiResults.M$FDR<0.05 & (CortArea_sensiResults.M$Outcome %in% CortArea.Sig.M), "ElementDescription"]`\
Prenatal vars:`r CortArea_sensiResults.M[CortArea_sensiResults.M$FDR<0.05 & (CortArea_sensiResults.M$Outcome %in% CortArea.Sig.M), "Predictor"]`\
\
\
**Cortical Sulcal Depth**
\
Out of `r length(SulcalDepth.Sig.F)` significant Sulcal depth regions associated with Prenatal Adversity in females, `r length(SulcalDepth_sensiResults.F[SulcalDepth_sensiResults.F$FDR<0.05 & (SulcalDepth_sensiResults.F$Outcome %in% SulcalDepth.Sig.F), "ElementDescription"])` were also significant.\
Overlaping ROIs:`r  SulcalDepth_sensiResults.F[SulcalDepth_sensiResults.F$FDR<0.05 & (SulcalDepth_sensiResults.F$Outcome %in% SulcalDepth.Sig.F), "ElementDescription"]`\
Prenatal vars:`r SulcalDepth_sensiResults.F[SulcalDepth_sensiResults.F$FDR<0.05 & (SulcalDepth_sensiResults.F$Outcome %in% SulcalDepth.Sig.F), "Predictor"]`\
\
Out of `r length(SulcalDepth.Sig.M)` significant Sulcal Depth regions associated with Prenatal Adversity in males, all `r length(SulcalDepth_sensiResults.M[SulcalDepth_sensiResults.M$FDR<0.05 & (SulcalDepth_sensiResults.M$Outcome %in% SulcalDepth.Sig.M), "Outcome"])`. were also significant\
Overlaping ROIs:`r names(table(SulcalDepth_sensiResults.M[SulcalDepth_sensiResults.M$FDR<0.05 & (SulcalDepth_sensiResults.M$Outcome %in% SulcalDepth.Sig.M), "Outcome"]))`\
Prenatal vars:`r names(table(SulcalDepth_sensiResults.M[SulcalDepth_sensiResults.M$FDR<0.05 & (SulcalDepth_sensiResults.M$Outcome %in% SulcalDepth.Sig.M), "Predictor"]))`\
\
\

# eFigure 2 - Correlations between study predictors
```{r, fig.width=10, fig.height=6}

R.f <- cor(abcd.f[, covars.x.labels$covars.x[1:22] ], use = 'complete.obs')
R.m <- cor(abcd.m[, covars.x.labels$covars.x[1:22] ], use = 'complete.obs')

## Merging MAtrices
corstarsM <- matrix(NA, nrow = 22, ncol =  22)
corstarsM[upper.tri(corstarsM)] <- R.f[upper.tri(R.f)]
corstarsM[lower.tri(corstarsM)] <- R.m[lower.tri(R.m)]

rownames(corstarsM) <- covars.x.labels$labels[1:22]
colnames(corstarsM) <- covars.x.labels$labels[1:22]

corrplot::corrplot(corstarsM, diag = F, addCoef.col = "black",  tl.col="black", tl.srt=45, tl.cex=.75,number.cex=.5 )

jpeg(filename = "eFigure2.jpg", height = 8,width = 12, units = 'in', res = 600)
corrplot::corrplot(corstarsM, diag = F, addCoef.col = "black",  tl.col="black", tl.srt=45, tl.cex=.75,number.cex=.5 )
dev.off()

```


# eFigure 3 - Population genetic stratification by reported ethnicity 

```{r, fig.width=10, fig.height=6}

for(i in 1:10){
    abcd[,"PC.y"] <- abcd[,paste0("PC",i)]
    assign(paste0("PC",i),
    abcd%>% dplyr::filter(!is.na(demo_race_ethnicity)) %>% 
        ggplot(aes(x= demo_race_ethnicity, y= PC.y))+
        geom_jitter(color= "black", size=0.2, alpha=0.4) +
        theme_bw()+
        geom_boxplot(aes(fill=demo_race_ethnicity),alpha=.7)+
        theme(
        legend.position="none",
        plot.title = element_text(size=11,hjust = 0.5),
        axis.text.x = element_text(size=9,angle = 45,hjust = 1)
    
        ) +
        labs(title = paste0("PC",i), x="",y="")
    )
}

gridExtra::grid.arrange(PC1,
                        PC2,
                        PC3,
                        PC4,
                        PC5,
                        PC6,
                        PC7,
                        PC8,
                        PC9,
                        PC10, nrow = 2)


jpeg(filename = "eFigure3.jpg", height = 6,width = 12, units = 'in', res = 600)
gridExtra::grid.arrange(PC1,
                        PC2,
                        PC3,
                        PC4,
                        PC5,
                        PC6,
                        PC7,
                        PC8,
                        PC9,
                        PC10, nrow = 2)

dev.off()

fit.Pop_strat <- summary(lm(smri_vol_cdk_total ~ PC1 + PC2 + PC3 , abcd))
cor_PGS_Brain <- stats::cor(abcd$smri_vol_cdk_total, abcd$brainvol.pgs,use = 'complete.obs')
# Multicollinarity check

VIf.formula <- paste("smri_vol_cdk_total ~ A_Prenatal_Burden_Score_z + ", paste(covars.x.labels$covars.x[-1], collapse = "+"))
fit.VIF     <- lm(as.formula(VIf.formula), data = abcd)
VIF.model   <- car::vif(fit.VIF)


```

Brain volume explained variance by the first three principal components of of population stratification `r round(fit.Pop_strat$r.squared, 2)`
Correlation between Brain volume PGS and total brain volume `r round(cor_PGS_Brain, 2)`


# eFigure 4
# eFigure 4A
## Principal Component Analysis of Significant Meta-Analysis Brain Regions
```{r }
# N significant Brain Regions
MetaSigROIsN.f <- sum(Meta_Betas.F$p.value<0.05)
MetaSigROIsN.m <- sum(Meta_Betas.M$p.value<0.05)

# Running Principal component analysis ####
# Females
MetaPCs.F                            <- principal(abcd.f[,Meta_Betas.F$ROI[Meta_Betas.F$p.value<0.05]], 3)
MetaPCsLoadings.F                    <- as.data.frame(MetaPCs.F$loadings[1:MetaSigROIsN.f, 1:3])
rownames(MetaPCsLoadings.F)          <- Meta_Betas.F$ElementDescription[Meta_Betas.F$p.value<0.05]
for(i in 1:3){MetaPCsLoadings.F[,i]  <- ifelse(abs(MetaPCsLoadings.F[,i])<.2,0, MetaPCsLoadings.F[,i])}

MetaPCsLoadings.F                    <- arrange(MetaPCsLoadings.F, desc(RC3))
MetaPCsLoadings.F                    <- arrange(MetaPCsLoadings.F, desc(RC2))
MetaPCsLoadings.F                    <- arrange(MetaPCsLoadings.F, desc(RC1))
MetaPCsLoadings.F$ElementDescription <- Meta_Betas.F$ElementDescription[Meta_Betas.F$p.value<0.05]

# Re-ordering Cols
colnames(MetaPCsLoadings.F)[1:3]     <- c('Meta Brain PC1', 'Meta Brain PC2','Meta Brain PC3')#renaming PCs
MetaPCsLoadings.F.long               <- reshape2::melt(MetaPCsLoadings.F,
                                             id="ElementDescription",
                                             measure=c('Meta Brain PC1', 'Meta Brain PC2','Meta Brain PC3'),
                                             variable.name="Brain components", value.name="Loading")

MetaPCsLoadings.F.long$ElementDescription <- factor(MetaPCsLoadings.F$ElementDescription,
                                                     levels = Meta_Betas.F$ElementDescription[Meta_Betas.F$p.value<0.05])

#Males
MetaPCs.M                            <- principal(abcd.m[,Meta_Betas.M$ROI[Meta_Betas.M$p.value<0.05]], 3)
MetaPCsLoadings.M                    <- as.data.frame(MetaPCs.M$loadings[1:MetaSigROIsN.m, 1:3])
rownames(MetaPCsLoadings.M)          <- Meta_Betas.M$ElementDescription[Meta_Betas.M$p.value<0.05]
for(i in 1:3){MetaPCsLoadings.M[,i]  <- ifelse(abs(MetaPCsLoadings.M[,i])<.2,0, MetaPCsLoadings.M[,i])}

MetaPCsLoadings.M                   <- arrange(MetaPCsLoadings.M, desc(RC3))
MetaPCsLoadings.M                   <- arrange(MetaPCsLoadings.M, desc(RC2))
MetaPCsLoadings.M                   <- arrange(MetaPCsLoadings.M, desc(RC1))
MetaPCsLoadings.M$ElementDescription<- Meta_Betas.M$ElementDescription[Meta_Betas.M$p.value<0.05]

# Re-ordering Cols
colnames(MetaPCsLoadings.M)[1:3]    <- c('Meta Brain PC2', 'Meta Brain PC1','Meta Brain PC3')#renaming PCs
MetaPCsLoadings.M.long              <- reshape2::melt(MetaPCsLoadings.M,
                                             id="ElementDescription",
                                             measure=c('Meta Brain PC1', 'Meta Brain PC2','Meta Brain PC3'),
                                             variable.name="Brain components", value.name="Loading")

MetaPCsLoadings.M.long$ElementDescription <- factor(MetaPCsLoadings.M$ElementDescription,
                                                     levels = Meta_Betas.M$ElementDescription[Meta_Betas.M$p.value<0.05])
```

## Create eFigure 4A - Factor lodings significant brain regions

```{r, fig.height=6, fig.width=12}

MetaLoadings.F <-
  ggplot(MetaPCsLoadings.F.long,aes(y = ElementDescription,x = round(Loading,1),abs(Loading),fill = Loading)) +
  facet_wrap( ~ `Brain components`, nrow = 1) +
  geom_bar(stat = "identity") +
  scale_fill_gradient2(name = "Loading", high = "red", mid = "white", low = "blue", midpoint = 0, guide = 'none') +
  theme_bw(base_size = 10)+
  geom_vline(xintercept = 0, linewidth=.25)+
  coord_cartesian(xlim = c(-.6,.8))+
    labs(y="Brain Data Modalities and Regions", title = 'Females')+
  theme(axis.text.y = element_text(size = 9),axis.text.x = element_text(size = 7.5),
          plot.title = element_text(size = 11,hjust = 0.5),axis.title.x =  element_blank())

MetaLoadings.M <-
  ggplot(MetaPCsLoadings.M.long,aes(y = ElementDescription,x = round(Loading,1),abs(Loading),fill = Loading)) +
  facet_wrap( ~ `Brain components`, nrow = 1) +
  geom_bar(stat = "identity") +
  scale_fill_gradient2(name = "Loading", high = "red", mid = "white", low = "blue", midpoint = 0, guide = 'none') +
  coord_cartesian(xlim = c(-.6,.8))+
  theme_bw(base_size = 10) +
  geom_vline(xintercept = 0, linewidth=.25)+
  labs(y="Brain Data Modalities and Regions", title = 'Males')+
  theme(axis.text.y = element_text(size = 9),axis.text.x = element_text(size = 7.5),
          plot.title = element_text(size = 11,hjust = 0.5),axis.title.x =  element_blank())

grid.arrange(MetaLoadings.F, MetaLoadings.M, ncol=2)
jpeg(filename = "eFigure4A.jpg", height = 6,width = 13, units = 'in', res = 600)
grid.arrange(MetaLoadings.F, MetaLoadings.M, ncol=2)
dev.off()

```

# eFigure 4B
## Analysis Meta Brain PCs and children's behavior and cognition

```{r Meta cbcl and brain PCs, cache=FALSE}

# Exploring PCs and CBCL and Cognitive measures

abcd.f[,'MetaBrain.PC1'] <-  as.numeric(MetaPCs.F$scores[,1])
abcd.f[,'MetaBrain.PC2'] <-  as.numeric(MetaPCs.F$scores[,2])
abcd.f[,'MetaBrain.PC3'] <-  as.numeric(MetaPCs.F$scores[,3])

abcd.m[,'MetaBrain.PC1'] <-  as.numeric(MetaPCs.M$scores[,2]) # Similar to PC1 loading patterns
abcd.m[,'MetaBrain.PC2'] <-  as.numeric(MetaPCs.M$scores[,1]) # Similar to PC2 loading patterns
abcd.m[,'MetaBrain.PC3'] <-  as.numeric(MetaPCs.M$scores[,3])

CBCL_results.F <- LME_loop(
  predictors = c('MetaBrain.PC1', 'MetaBrain.PC2', "MetaBrain.PC3"),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.f
)

CBCL_results.F$FDR <- p.adjust(CBCL_results.F$p_value_Predictor, 'fdr')

CBCL_results.M <- LME_loop(
  predictors = c('MetaBrain.PC1', 'MetaBrain.PC2', "MetaBrain.PC3"),
  outcomes = c(
    'cbcl_scr_syn_external_t',
    "cbcl_scr_syn_internal_t",
    "cbcl_scr_syn_totprob_t",
    'nihtbx_totalcomp_fc',# Cognition Total Composite Score Fully-Corrected z-score
    'nihtbx_cryst_fc',    # Crystallized Composite Fully-Corrected z-score 
    'nihtbx_fluidcomp_fc' # Cognition Fluid Composite Fully-Corrected z-score
  ),
  covars = covars.x,
  Scale = TRUE,
  data = abcd.m
)

CBCL_results.M$FDR <- p.adjust(CBCL_results.M$p_value_Predictor, 'fdr')

```

## Associations Between Meta-analysis Brain Components with Behavior Problems and Cognition

```{r,Meta CBCL_results.F}

kable(cbind(CBCL_results.F[,-c(1,3:4)],CBCL_results.M[,-c(1:5)]), 
      digits = 4, 
      caption = "Associations Meta-analysis Brain Components with Behavior Problems and Cognition",row.names = F)%>%
  kableExtra::kable_styling(position = "center")%>%
  add_header_above(c(" "=2, 'Females'=6, 'Males'=6))

```


```{r}
kable(cbind(Prenatal_results.F[,-c(1,3:4)],Prenatal_results.M[,-c(1:5)]), 
      digits = 3, 
      caption = "Associations Prenatal Adverse Exposures with Behavior Problems and Cognition",row.names = F)%>%
  kableExtra::kable_styling(position = "center")%>%add_header_above(c(" "=2, 'Females'=6, 'Males'=6))

```


## Specifing mediation model of Meta-analysis Brain PCs

```{r Mediation Meta-analysis Pcs, cache=FALSE}

#### Specifing Mediation Models ####
# Mediation MetaBrain PC1 ####
#Total problems
med.MetaBrainPC1.Tot <- "
    # direct effect
    cbcl_scr_syn_totprob_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_totprob_t.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_totprob_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

#Externalizing problems - PC1
med.MetaBrainPC1.Ext <- "
    # direct effect
    cbcl_scr_syn_external_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_external_t.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2 ~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_external_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"
# Internalizing
med.MetaBrainPC1.Int <- "
    # direct effect
    cbcl_scr_syn_internal_t.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    cbcl_scr_syn_internal_t.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    cbcl_scr_syn_internal_t.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

# Total Cognition
med.MetaBrainPC1.CogTotal <- "
    # direct effect
    nihtbx_totalcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_totalcomp_fc.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_totalcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.MetaBrainPC2.CogTotal <-  "
    # direct effect
    nihtbx_totalcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC2 ~ a*A_Prenatal_Burden_Score
    nihtbx_totalcomp_fc.std ~ b*MetaBrain.PC2
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_totalcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"


# Fluid Cognition - PC1 and PC2
med.MetaBrainPC1.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.MetaBrainPC2.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC2 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*MetaBrain.PC2
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3 ~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

med.MetaBrainPC3.CogFluid <- "
    # direct effect
    nihtbx_fluidcomp_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    
    MetaBrain.PC3 ~ a*A_Prenatal_Burden_Score
    nihtbx_fluidcomp_fc.std ~ b*MetaBrain.PC3
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ aPC1*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2 ~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3 ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_fluidcomp_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

# Cryst Cognition
med.MetaBrainPC1.CogCryst <- "
    # direct effect
    nihtbx_cryst_fc.std ~ c*A_Prenatal_Burden_Score
    # mediator
    MetaBrain.PC1 ~ a*A_Prenatal_Burden_Score
    nihtbx_cryst_fc.std ~ b*MetaBrain.PC1
    
    # indirect effect (a*b)
    ab := a*b
    # total effect
    total := c + (a*b)
    #Indirect/total effect
    abtot := ab/total

    #Covars
    MetaBrain.PC1~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC2~ aPC2*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    MetaBrain.PC3~ aPC3*A_Prenatal_Burden_Score + interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness
    
    nihtbx_cryst_fc.std ~ interview_age_z + demo_comb_income_v2_z +demo_caregiver_ed_z + fes_fam_env_score_z + demo_prnt_marital + devhx_3_p_z +
    asr_scr_totprob_t_z + neighborhood_safety_z + Average_Screentime_z + Exp_trauma + PC1_z +PC2_z + PC3_z +PC4_z + PC5_z + PC6_z + PC7_z+
    PC8_z + PC9_z + PC10_z + brainvol.pgs_z + Handedness

"

#Mediation Females

fit.MetaBrainPC1.CogTotal.f <- lavaan::sem(med.MetaBrainPC1.CogTotal, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.MetaBrainPC1.CogFluid.f <- lavaan::sem(med.MetaBrainPC1.CogFluid, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.MetaBrainPC1.CogCryst.f <- lavaan::sem(med.MetaBrainPC1.CogCryst, data = abcd.f, se = "bootstrap", bootstrap = 1000)

fit.MetaBrainPC1.Ext.f      <- lavaan::sem(med.MetaBrainPC1.Ext, data = abcd.f, se = "bootstrap", bootstrap = 1000)

fit.MetaBrainPC2.CogFluid.f <- lavaan::sem(med.MetaBrainPC2.CogFluid, data = abcd.f, se = "bootstrap", bootstrap = 1000)
fit.MetaBrainPC2.CogTotal.f <- lavaan::sem(med.MetaBrainPC2.CogTotal, data = abcd.f, se = "bootstrap", bootstrap = 1000)

#Mediation Males
fit.MetaBrainPC1.Tot.m      <- lavaan::sem(med.MetaBrainPC1.Tot     , data = abcd.m, se = "bootstrap", bootstrap = 1000)
fit.MetaBrainPC1.Ext.m      <- lavaan::sem(med.MetaBrainPC1.Ext     , data = abcd.m, se = "bootstrap", bootstrap = 1000)

```

## Preparing Mediation Results
```{r}
#Females
fit.Pars.MetaBrainPC1.CogTotal.f <- lavaan::parameterestimates(fit.MetaBrainPC1.CogTotal.f, boot.ci.type = "bca.simple") 
fit.Pars.MetaBrainPC1.CogFluid.f <- lavaan::parameterestimates(fit.MetaBrainPC1.CogFluid.f, boot.ci.type = "bca.simple") 
fit.Pars.MetaBrainPC1.CogCryst.f <- lavaan::parameterestimates(fit.MetaBrainPC1.CogCryst.f, boot.ci.type = "bca.simple") 

fit.Pars.PC1.Ext.f           <- lavaan::parameterestimates(fit.MetaBrainPC1.Ext.f, boot.ci.type = "bca.simple") 

fit.Pars.MetaBrainPC2.CogFluid.f <- lavaan::parameterestimates(fit.MetaBrainPC2.CogFluid.f, boot.ci.type = "bca.simple") 
fit.Pars.MetaBrainPC2.CogTotal.f <- lavaan::parameterestimates(fit.MetaBrainPC2.CogTotal.f, boot.ci.type = "bca.simple") 

# Males
fit.Pars.MetaBrainPC1.Tot.m      <- lavaan::parameterestimates(fit.MetaBrainPC1.Tot.m, boot.ci.type = "bca.simple") 
fit.Pars.MetaBrainPC1.Ext.m      <- lavaan::parameterestimates(fit.MetaBrainPC1.Ext.m, boot.ci.type = "bca.simple")

```

## Create eFigure 4B - Mediation meta-analysis parameters used in eFigure 4B

## Mediation Parameters Females
```{r}
kable(fit.Pars.MetaBrainPC1.CogTotal.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Total Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.MetaBrainPC1.CogFluid.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.MetaBrainPC1.CogCryst.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")


kable(fit.Pars.PC1.Ext.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC1 for Externalizing Problems  - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.MetaBrainPC2.CogFluid.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC2 for Fluid Cognition - Females") %>% 
  kable_styling(position = "center")

kable(fit.Pars.MetaBrainPC2.CogTotal.f %>% filter(label != ""), digits = 3, caption = "Mediation parameters PC2 for Total Cognition - Females") %>% 
  kable_styling(position = "center")



```

## Mediation Parameters Males


```{r}

kable(fit.Pars.MetaBrainPC1.Tot.m %>% filter(label!=""), digits = 3,caption = "Mediation parameters for Total Problems - Males") %>% 
  kable_styling(position = "center")

kable(fit.Pars.MetaBrainPC1.Ext.m %>% filter(label!=""), digits = 3,caption = "Mediation parameters for Externalizing Problems - Males") %>% 
  kable_styling(position = "center")


```

# eFigure 5
# Sensitivity analysis manual quality control
## Getting significant ROIs and brain structure covar
```{r}
Main_sig_results.F <- Main_results.F[Main_results.F$FDR<0.05,]
Main_sig_results.M <- Main_results.M[Main_results.M$FDR<0.05,]

rownames(Main_sig_results.F) <- 1:nrow(Main_sig_results.F)
rownames(Main_sig_results.M) <- 1:nrow(Main_sig_results.M)
# getting data modality
extract_character2 <- function(x) {
  words <- stringr::str_split(x, " ", simplify = TRUE)
  first_two_words <- paste(words[1, 1], words[1, 2])
  return(first_two_words)
}

Main_sig_results.F$Type <- sapply(Main_sig_results.F$ElementDescription, extract_character2)
Main_sig_results.M$Type <- sapply(Main_sig_results.M$ElementDescription, extract_character2)


Main_sig_results.F$Brain_covar <- car::recode(Main_sig_results.F$Type,
                                              "'Cortical volume'= 'smri_vol_scs_intracranialv_z' ;
                                              'Subcortical volume' = 'smri_vol_scs_intracranialv';
                                              'Cortical thickness'= 'smri_thick_cdk_mean_z';
                                              'Cortical area'= 'smri_area_cdk_total_z';
                                              'Sulcal depth'= 'smri_sulc_cdk_mean_z'")

Main_sig_results.M$Brain_covar <- car::recode(Main_sig_results.M$Type,
                                              "'Cortical volume'= 'smri_vol_scs_intracranialv_z' ;
                                              'Subcortical volume' = 'smri_vol_scs_intracranialv';
                                              'Cortical thickness'= 'smri_thick_cdk_mean_z';
                                              'Cortical area'= 'smri_area_cdk_total_z';
                                              'Sulcal depth'= 'smri_sulc_cdk_mean_z'")


```

## Runnning analysis

```{r QC analysis, warning=FALSE}
Main_sig_results_manualQC.F <- Main_sig_results.F[0,]
Main_sig_results_manualQC.M <- Main_sig_results.M[0,]

for( i in 1:nrow(Main_sig_results.F)){
  results_manualQC.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.F[i,"Outcome"],
    outcomesLab = Main_sig_results.F[i,"Label"],
    ROI = Main_sig_results.F[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x,Main_sig_results.F[i,"Brain_covar"]),
    data = abcd.f[abcd.f$imgincl_t1w_include==1,]
  )
  
  
  Main_sig_results_manualQC.F <- rbind(Main_sig_results_manualQC.F, results_manualQC.F)
  
}

for( i in 1:nrow(Main_sig_results.M)){
  results_manualQC.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.M[i,"Outcome"],
    outcomesLab = Main_sig_results.M[i,"Label"],
    ROI = Main_sig_results.M[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x,Main_sig_results.M[i,"Brain_covar"]),
    data = abcd.m[abcd.m$imgincl_t1w_include==1,]
  )
  
  Main_sig_results_manualQC.M <- rbind(Main_sig_results_manualQC.M, results_manualQC.M)
  
}



```

## Preparing results
```{r}
conf_interval <- function(beta, se, conf_level = 0.95) {
  z <- qnorm((1 + conf_level) / 2)
  lower_bound <- beta - z * se
  upper_bound <- beta + z * se
  
  data.frame(
    beta = beta,
    se = se,
    lower_bound = lower_bound,
    upper_bound = upper_bound,
    conf_level = conf_level
  )
}

# Females
# Original
x <- conf_interval(beta = Main_sig_results.F$Beta_Predictor, se = Main_sig_results.F$Std.error, conf_level = .99)
Main_sig_results.F$lower_bound <- x$lower_bound
Main_sig_results.F$upper_bound <- x$upper_bound

# Manual QC
x <- conf_interval(beta = Main_sig_results_manualQC.F$Beta_Predictor, se = Main_sig_results_manualQC.F$Std.error, conf_level = .99)
Main_sig_results_manualQC.F$lower_bound <- x$lower_bound
Main_sig_results_manualQC.F$upper_bound <- x$upper_bound

# Males
# Original
x <- conf_interval(beta = Main_sig_results.M$Beta_Predictor, se = Main_sig_results.M$Std.error, conf_level = .99)
Main_sig_results.M$lower_bound <- x$lower_bound
Main_sig_results.M$upper_bound <- x$upper_bound

# Manual QC
x <- conf_interval(beta = Main_sig_results_manualQC.M$Beta_Predictor, se = Main_sig_results_manualQC.M$Std.error, conf_level = .99)
Main_sig_results_manualQC.M$lower_bound <- x$lower_bound
Main_sig_results_manualQC.M$upper_bound <- x$upper_bound



Main_sig_results_manualQC.F$QC <- "Manual"
Main_sig_results.F$QC          <- "Automated"

sig_results_QCcomparison.F     <- rbind(
  Main_sig_results_manualQC.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","QC")],
  Main_sig_results.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","QC")]
  )

Main_sig_results_manualQC.M$QC <- "Manual"
Main_sig_results.M$QC          <- "Automated"

sig_results_QCcomparison.M     <- rbind(
  Main_sig_results_manualQC.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","QC")],
  Main_sig_results.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","QC")]
  )

```

## Create eFigure 5

```{r, message=FALSE, warning=FALSE, fig.height=6*1.2, fig.width=10*1.2}

r.cor.F <- cor(Main_sig_results_manualQC.F$Beta_Predictor, Main_sig_results.F$Beta_Predictor)
plotQC.F <- ggplot(sig_results_QCcomparison.F, aes(y=ElementDescription, x=Beta_Predictor, col=QC))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = 'none')+
  labs(title = 'Female',
       subtitle = paste("Correlation estimated betas: \nr=", round(r.cor.F, 5)),
       y=""
       )

r.cor.M <- cor(Main_sig_results_manualQC.M$Beta_Predictor, Main_sig_results.M$Beta_Predictor)
plotQC.M <- ggplot(sig_results_QCcomparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=QC))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  labs(title = 'Male',
       subtitle = paste("r=", round(r.cor.M, 5)),
      y=""
       )


my_legend <-  cowplot::get_legend (plotQC.M)

plotQC.M <- ggplot(sig_results_QCcomparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=QC))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = 'none')+
  labs(title = 'Male',
       subtitle = paste("Correlation estimated betas: \nr=", round(r.cor.M, 5)),
      y=""
       )



grid.arrange(
  plotQC.F,
  plotQC.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

jpeg(filename = "eFigure5.jpg", height = 6*1.2,width = 10*1.2, units = 'in', res = 600)
grid.arrange(
  plotQC.F,
  plotQC.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

dev.off()


```



# eFigure 6
# Sensitivity analysis population stratification
## Runnning analysis

```{r Pop strat analysis and ethinicity, warning=FALSE}
Main_sig_results_PopStrat.F  <- Main_sig_results.F[0,]
Main_sig_results_PopStrat.M  <- Main_sig_results.M[0,]

for( i in 1:nrow(Main_sig_results.F)){
  
  results_PopStrat.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.F[i,"Outcome"],
    outcomesLab = Main_sig_results.F[i,"Label"],
    ROI = Main_sig_results.F[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x[c(1:10,22)],'demo_race_ethnicity',Main_sig_results.F[i,"Brain_covar"]),
    data = abcd.f
  )
  
  
  Main_sig_results_PopStrat.F <- rbind(Main_sig_results_PopStrat.F, results_PopStrat.F)
  
}

for( i in 1:nrow(Main_sig_results.M)){
  results_PopStrat.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.M[i,"Outcome"],
    outcomesLab = Main_sig_results.M[i,"Label"],
    ROI = Main_sig_results.M[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x[c(1:10,22)],'demo_race_ethnicity',Main_sig_results.M[i,"Brain_covar"]),
    data = abcd.m
  )
  
  Main_sig_results_PopStrat.M <- rbind(Main_sig_results_PopStrat.M, results_PopStrat.M)
  
}



```

## Preparing results
```{r}
# Females
# Original
x <- conf_interval(beta = Main_sig_results.F$Beta_Predictor, se = Main_sig_results.F$Std.error, conf_level = .99)
Main_sig_results.F$lower_bound <- x$lower_bound
Main_sig_results.F$upper_bound <- x$upper_bound

# No Population stratification
x <- conf_interval(beta = Main_sig_results_PopStrat.F$Beta_Predictor, se = Main_sig_results_PopStrat.F$Std.error, conf_level = .99)
Main_sig_results_PopStrat.F$lower_bound <- x$lower_bound
Main_sig_results_PopStrat.F$upper_bound <- x$upper_bound

Main_sig_results_PopStrat.F$Ancestry <- "Reported ethnicity"
Main_sig_results.F$Ancestry          <- "Genetic PCs"

sig_results_PopStrat_comparison.F     <- rbind(
  Main_sig_results_PopStrat.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","Ancestry")],
  Main_sig_results.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","Ancestry")]
  )

# Males
# Original
x <- conf_interval(beta = Main_sig_results.M$Beta_Predictor, se = Main_sig_results.M$Std.error, conf_level = .99)
Main_sig_results.M$lower_bound <- x$lower_bound
Main_sig_results.M$upper_bound <- x$upper_bound

# No Population stratification
x <- conf_interval(beta = Main_sig_results_PopStrat.M$Beta_Predictor, se = Main_sig_results_PopStrat.M$Std.error, conf_level = .99)
Main_sig_results_PopStrat.M$lower_bound <- x$lower_bound
Main_sig_results_PopStrat.M$upper_bound <- x$upper_bound

Main_sig_results_PopStrat.M$Ancestry <- "Reported ethnicity"
Main_sig_results.M$Ancestry          <- "Genetic PCs"

sig_results_PopStrat_comparison.M     <- rbind(
  Main_sig_results_PopStrat.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound",'Ancestry')],
  Main_sig_results.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound",'Ancestry')]
  )

```

## Create eFigure 6

```{r, message=FALSE, warning=FALSE, fig.height=6*1.2, fig.width=10*1.12 }

r.cor.F <- cor(Main_sig_results_PopStrat.F$Beta_Predictor, Main_sig_results.F$Beta_Predictor)
plotPop.F <- ggplot(sig_results_PopStrat_comparison.F, aes(y=ElementDescription, x=Beta_Predictor, col=Ancestry))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = 'none')+
  labs(title = 'Female',
       subtitle = paste("Correlation estimated betas:\n r=", round(r.cor.F, 5)),
       y=""
       )

r.cor.M <- cor(Main_sig_results_PopStrat.M$Beta_Predictor, Main_sig_results.M$Beta_Predictor)
plotPop.M <- ggplot(sig_results_PopStrat_comparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=Ancestry))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  labs(title = 'Male',
       subtitle = paste("Correlation estimated betas:\n r=", round(r.cor.M, 5)),
      y=""
       )
my_legend <- cowplot::get_legend(plotPop.M)
plotPop.M <- ggplot(sig_results_PopStrat_comparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=Ancestry))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = "none")+
  labs(title = 'Male',
       subtitle = paste("Correlation estimated betas:\n r=", round(r.cor.M, 5)),
      y=""
       )


grid.arrange(
  plotPop.F,
  plotPop.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

jpeg(filename = "eFigure6.jpg", height = 6*1.2,width = 10*1.2, units = 'in', res = 600)
grid.arrange(
  plotQC.F,
  plotQC.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

dev.off()


```

# eFigure7
# Sensitivety analysis for postnatal exposure to Tobbaco
## Running analysis
```{r tobbaco sensitivity,warning=FALSE}

Main_sig_results_Tobacco.F  <- Main_sig_results.F[0,]
Main_sig_results_Tobacco.M  <- Main_sig_results.M[0,]

for( i in 1:nrow(Main_sig_results.F)){
  
  results_Tobacco.F <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.F[i,"Outcome"],
    outcomesLab = Main_sig_results.F[i,"Label"],
    ROI = Main_sig_results.F[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x,'scrn_hr_smoke',Main_sig_results.F[i,"Brain_covar"]),
    data = abcd.f
  )
  
  
  Main_sig_results_Tobacco.F <- rbind(Main_sig_results_Tobacco.F, results_Tobacco.F)
  
}

for( i in 1:nrow(Main_sig_results.M)){
  
  results_Tobacco.M <-
  LME_loop(
    predictors = "A_Prenatal_Burden_Score_z",
    outcomes =  Main_sig_results.M[i,"Outcome"],
    outcomesLab = Main_sig_results.M[i,"Label"],
    ROI = Main_sig_results.M[i,"ElementDescription"],
    Scale = T,
    covars = c(covars.x,'scrn_hr_smoke',Main_sig_results.M[i,"Brain_covar"]),
    data = abcd.m
  )
  
  
  Main_sig_results_Tobacco.M <- rbind(Main_sig_results_Tobacco.M, results_Tobacco.M)
  
}

```

## Preparing results
```{r}
# Females
# Original
x <- conf_interval(beta = Main_sig_results.F$Beta_Predictor, se = Main_sig_results.F$Std.error, conf_level = .99)
Main_sig_results.F$lower_bound <- x$lower_bound
Main_sig_results.F$upper_bound <- x$upper_bound

# Postnatal smoking
x <- conf_interval(beta = Main_sig_results_Tobacco.F$Beta_Predictor, se = Main_sig_results_Tobacco.F$Std.error, conf_level = .99)
Main_sig_results_Tobacco.F$lower_bound <- x$lower_bound
Main_sig_results_Tobacco.F$upper_bound <- x$upper_bound

Main_sig_results_Tobacco.F$Postnatal_tobacco_exposure <- "Yes"
Main_sig_results.F$Postnatal_tobacco_exposure <- "No"

Main_sig_results_Tobacco_comparison.F     <- rbind(
  Main_sig_results_Tobacco.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","Postnatal_tobacco_exposure")],
  Main_sig_results.F[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound","Postnatal_tobacco_exposure")]
  )

# Males
# Original
x <- conf_interval(beta = Main_sig_results.M$Beta_Predictor, se = Main_sig_results.M$Std.error, conf_level = .99)
Main_sig_results.M$lower_bound <- x$lower_bound
Main_sig_results.M$upper_bound <- x$upper_bound

# Postnatal smoking
x <- conf_interval(beta = Main_sig_results_Tobacco.M$Beta_Predictor, se = Main_sig_results_Tobacco.M$Std.error, conf_level = .99)
Main_sig_results_Tobacco.M$lower_bound <- x$lower_bound
Main_sig_results_Tobacco.M$upper_bound <- x$upper_bound

Main_sig_results_Tobacco.M$Postnatal_tobacco_exposure <- "Yes"
Main_sig_results.M$Postnatal_tobacco_exposure <- "No"

Main_sig_results_Tobacco_comparison.M     <- rbind(
  Main_sig_results_Tobacco.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound",'Postnatal_tobacco_exposure')],
  Main_sig_results.M[,c("ElementDescription", "Beta_Predictor","lower_bound","upper_bound",'Postnatal_tobacco_exposure')]
  )

```

## Create eFigure 7

```{r, message=FALSE, warning=FALSE, fig.height=6*1.2, fig.width=10*1.2}

r.cor.F       <- cor(Main_sig_results_Tobacco.F$Beta_Predictor, Main_sig_results.F$Beta_Predictor)
plotTobacco.F <- ggplot(Main_sig_results_Tobacco_comparison.F, aes(y=ElementDescription, x=Beta_Predictor, col=Postnatal_tobacco_exposure))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = 'none')+
  labs(title = 'Female',
       subtitle = paste("Correlation estimated betas: \nr=", round(r.cor.F, 5)),
       y=""
       )

r.cor.M       <- cor(Main_sig_results_Tobacco.M$Beta_Predictor, Main_sig_results.M$Beta_Predictor)
plotTobacco.M <- ggplot(Main_sig_results_Tobacco_comparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=Postnatal_tobacco_exposure))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  labs(title = 'Male',
       subtitle = paste("r=", round(r.cor.M, 5)),
      y=""
       )

my_legend <-  cowplot::get_legend (plotTobacco.M)

plotTobacco.M <- ggplot(Main_sig_results_Tobacco_comparison.M, aes(y=ElementDescription, x=Beta_Predictor, col=Postnatal_tobacco_exposure))+
  geom_point(position = position_dodge(width = 0.25))+
  geom_errorbarh(aes(xmin = lower_bound, xmax = upper_bound), height = 0.2, position = position_dodge(width = 0.25))+
  scale_color_manual(values = c('#9EB4D3','#FFBE98'))+
  theme_bw()+
  theme(legend.position = "none")+
  labs(title = 'Male',
       subtitle = paste("r=", round(r.cor.M, 5)),
      y=""
       )


grid.arrange(
  plotTobacco.F,
  plotTobacco.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

jpeg(filename = "eFigure7.jpg", height = 6*1.2,width = 10*1.2, units = 'in', res = 600)
grid.arrange(
  plotQC.F,
  plotQC.M,
  my_legend,
  heights = c(10, 1),layout_matrix = rbind(c(1,1,2,2), c(NA,3,3,NA)),
  ncol = 2
)

dev.off()


```



```{r, echo=FALSE, eval=FALSE}
save.image(file = "/data1/meaneylab/euclides/3_ABCD cohort/Prenatal_Adversity_Univariate/Analysis_Update/PrenatalAdversity.RData")
load("/data1/meaneylab/euclides/3_ABCD cohort/Prenatal_Adversity_Univariate/PrenatalAdversity.RData")

#abcd,
#abcd.f,
#abcd.m,
#CortVolResults.M,
#CortVolResults.F,
#CortThickResults.F,
#CortThickResults.M,
#CortAreaResults.M,
#CortAreaResults.F,
```



```{r, echo=FALSE}
#markdownToHTML('MethylationQC_HERO_Cleand_and_Normalisation.md', 'MethylationQC_HERO_Cleand_and_Normalisation.html')
#markdown::mark_latex('MethylationQC_HERO_Cleand_and_Normalisation.md','MethylationQC_HERO_Cleand_and_Normalisation.pdf')

```
